

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Neuromorphological File Specification &mdash; Neuromophological File Specification 4.0.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Release Notes" href="Release.html" />
    <link rel="prev" title="Welcome to the Neuromorphological File Specification" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> Neuromophological File Specification
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Neuromorphological File Specification</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#definitions">1. Definitions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#introduction-to-the-neuromorphological-file-format">2. Introduction to the Neuromorphological File Format</a></li>
<li class="toctree-l2"><a class="reference internal" href="#neuromorphological-file-contents">3. Neuromorphological File Contents</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#coordinate-space">3.1 Coordinate Space</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#header-elements">4. Header Elements</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#description">4.1 &lt;description&gt;</a></li>
<li class="toctree-l3"><a class="reference internal" href="#filefacts">4.2 &lt;filefacts&gt;</a></li>
<li class="toctree-l3"><a class="reference internal" href="#sparcdata">4.3 &lt;sparcdata&gt;</a></li>
<li class="toctree-l3"><a class="reference internal" href="#property-name-timepointmanager">4.4 &lt;property name=”TimePointManager”&gt;</a></li>
<li class="toctree-l3"><a class="reference internal" href="#images">4.5 &lt;images&gt;</a></li>
<li class="toctree-l3"><a class="reference internal" href="#thumbnail">4.6 &lt;thumbnail&gt;</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#trace-data">5.Trace Data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#common-components">5.1 Common Components</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#color">5.1.1 Color</a></li>
<li class="toctree-l4"><a class="reference internal" href="#point">5.1.2 &lt;point&gt;</a></li>
<li class="toctree-l4"><a class="reference internal" href="#property-name-channel">5.1.3 &lt;property name=”Channel”&gt;</a></li>
<li class="toctree-l4"><a class="reference internal" href="#property-name-set">5.1.4 &lt;property name=”Set”&gt;</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#marker">5.2 &lt;marker&gt;</a></li>
<li class="toctree-l3"><a class="reference internal" href="#punctum">5.3 Punctum</a></li>
<li class="toctree-l3"><a class="reference internal" href="#contour">5.4 &lt;contour&gt;</a></li>
<li class="toctree-l3"><a class="reference internal" href="#neurons">5.5 Neurons</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#tree">5.5.1 &lt;tree&gt;</a></li>
<li class="toctree-l4"><a class="reference internal" href="#somas-and-cell-bodies">5.5.2 Somas and Cell Bodies</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#vascular-networks">5.6 Vascular Networks</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#vessel">5.6.1 &lt;vessel&gt;</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#annotations">5.7 Annotations</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#arrow">5.7.1 &lt;arrow&gt;</a></li>
<li class="toctree-l4"><a class="reference internal" href="#text">5.7.2 &lt;text&gt;</a></li>
<li class="toctree-l4"><a class="reference internal" href="#scalebar">5.7.3 &lt;scalebar&gt;</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#references">6. References</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Release.html">Release Notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="Credits.html">Credits</a></li>
<li class="toctree-l1"><a class="reference internal" href="Legal.html">Legal</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Neuromophological File Specification</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Neuromorphological File Specification</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/NMF.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="neuromorphological-file-specification">
<h1>Neuromorphological File Specification<a class="headerlink" href="#neuromorphological-file-specification" title="Permalink to this headline">¶</a></h1>
<p>Version 1.0.0 <a class="footnote-reference brackets" href="#id3" id="id1">1</a></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For additional information, visit MBF Bioscience’s user forum for the <a class="reference external" href="https://forums.mbfbioscience.com/c/neurolucida-users/neurolucida-360/19">Neuromorphological File Specification.</a></p>
</div>
<div class="section" id="definitions">
<h2>1. Definitions<a class="headerlink" href="#definitions" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li><p>Research Resource Identifier (RRID): a persistent and unique identifier for referencing a research resource.</p></li>
</ol>
</div>
<div class="section" id="introduction-to-the-neuromorphological-file-format">
<h2>2. Introduction to the Neuromorphological File Format<a class="headerlink" href="#introduction-to-the-neuromorphological-file-format" title="Permalink to this headline">¶</a></h2>
<p>This specification describes the elements of the neuromorphological data file. The structures represented have been continually developed for over 30 years, balancing accurate 3D representation of microscopic structures with the efficiency for reading, writing, storing, and rendering the reconstruction data. The neuromorphological file format balances structure with flexibility by storing each modeled object as a unique data element, and providing mechanisms for grouping any number and type of data elements. File-level metadata is stored that provides detail on the sample origin to ensure no important source information is separated from the corresponding data. Each structure implemented has had a strong influence from top neuroscientists to maintain a meaningful model and provide the necessary analysis options for that entity. The file specification will continue to be updated as needed to define added or modified data elements.</p>
</div>
<div class="section" id="neuromorphological-file-contents">
<h2>3. Neuromorphological File Contents<a class="headerlink" href="#neuromorphological-file-contents" title="Permalink to this headline">¶</a></h2>
<p>The neuromorphological file format is an Extensible Markup Language (XML) 1.0 (Fifth Edition) format and includes two organizational aspects, elements, and attributes [1]. These aspects are detailed in the neuromorphological file specification provided by The World Wide Web Consortium (WC3). All neuromorphological data files have a header section that includes the file’s introductory content followed by any traced data elements. Traced data elements are representations of a diverse range of neuromorphological structures found in microscopy image data.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="cp">&lt;?xml version=&quot;1.0&quot; encoding=&quot;ISO-8859-1&quot;?&gt;</span>
<span class="nt">&lt;mbf</span> <span class="na">version=</span> <span class="s">4.0</span> <span class="na">appname=</span><span class="s">&quot;ApplicationName&quot;</span> <span class="na">appversion=</span><span class="s">&quot;YYYY.V.M&quot;</span> <span class="na">apprrid=</span><span class="s">&quot;SCR_xxxxx&quot;</span> <span class="na">insrrid=</span><span class="s">&quot;SCR_xxxxx&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;description&gt;&lt;/description&gt;</span>
  <span class="nt">&lt;filefacts&gt;&lt;/filefacts&gt;</span>
  <span class="nt">&lt;sparcdata&gt;&lt;/sparcdata&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;TimePointManager&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;images&gt;&lt;/images&gt;</span>
  <span class="nt">&lt;thumbnail</span> <span class="na">cols=</span><span class="s">&quot;64&quot;</span> <span class="na">rows=</span><span class="s">&quot;64&quot;</span><span class="nt">&gt;&lt;/thumbnail&gt;</span>

  TRACING DATA

<span class="nt">&lt;/mbf&gt;</span>
</pre></div>
</div>
<p><em>Figure 1 A Neuromorphological data file.  The header elements are displayed in blue. All MBF tracing data is a nested element of the &lt;mbf&gt; element and is represented above with the comment, TRACING DATA.</em></p>
<div class="section" id="coordinate-space">
<h3>3.1 Coordinate Space<a class="headerlink" href="#coordinate-space" title="Permalink to this headline">¶</a></h3>
<p>The coordinate space for all neuromorphological data files is a three-dimensional space and all coordinates and measurements are in micrometer units (µm). The origin point of the coordinate system is (0, 0, and 0).</p>
<a class="reference internal image-reference" href="_images/1.png"><img alt="" class="align-center" src="_images/1.png" style="width: 400px; height: 400px;" /></a>
<p><em>Figure 2 demonstrates the 3D coordinate space with an origin point of (0, 0, and 0). The gray planes represent a 3D image volume with an image location coordinate, coord: (x, y, and z). Note the direction of the Z-axis. The most positive image plane of the 3D volume is image plane 1. The following image planes are in the same X and Y location, but their z location changes incrementally based on the z scaling. The units of this coordinate space are in micrometers (µm).</em></p>
</div>
</div>
<div class="section" id="header-elements">
<h2>4. Header Elements<a class="headerlink" href="#header-elements" title="Permalink to this headline">¶</a></h2>
<p>The first line of all neuromorphological data files is always the XML declaration element. It contains two attributes, version and encoding, that define the structure and storage units of the neuromorphological file format.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="cp">&lt;?xml version=&quot;1.0&quot; encoding=&quot;ISO-8859-1&quot;?&gt;</span>
</pre></div>
</div>
<p><em>Figure 3 The XML document declaration element, attributes, and values as they apear in the segmentation data file.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 16%" />
<col style="width: 68%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Element</p></th>
<th class="head"><p>Attributes</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;?&gt;</p></td>
<td><p>The XML declaration element informs the structure
and storage units of the document.</p></td>
</tr>
<tr class="row-odd"><td rowspan="2"></td>
<td><p>version</p></td>
<td><p>The XML document specification version number.
For all segmentation files created with MBF
Bioscience software, this value will equal “1.0”
indicating the version of the XML specification
[1].</p></td>
</tr>
<tr class="row-even"><td><p>encoding</p></td>
<td><p>Identifies the character format in the file.
This value will equal “ISO-8859-1” for all
segmentation files created with MBF Bioscience
software, implying single-byte encoding [2].</p></td>
</tr>
</tbody>
</table>
<p><em>Table 1 The XML declaration element, attributes, and descriptions for all fields.</em></p>
<p>The remainder of the data file is embedded within the &lt;mbf&gt; element, including file header information and the elements that comprise the tracing data used to model neuronal morphology and surrounding anatomies. The closing &lt;/mbf&gt; is the last line of all neuromorphological data files.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;mbf</span> <span class="na">version=</span> <span class="s">4.0</span> <span class="na">appname=</span><span class="s">&quot;ApplicationName&quot;</span> <span class="na">appversion=</span><span class="s">&quot;YYYY.V.M&quot;</span> <span class="na">apprrid=</span><span class="s">&quot;SCR_xxxxx&quot;</span> <span class="na">insrrid=</span><span class="s">&quot;SCR_xxxxx&quot;</span><span class="nt">&gt;</span>
  ⋮
<span class="nt">&lt;/mbf&gt;</span>
</pre></div>
</div>
<p><em>Figure 4 The &lt;mbf&gt; element, attributes, and values as they apear in the segmentation data file. The appversion for all MBF Bioscience software is reported with the year, version, and minor version (YYYY.V.M). The virtical elipses between the attributes of the &lt;mbf&gt; elements  and the &lt;/mbf&gt; end tag is used as shorthand to indicate there are additional child elements of the &lt;mbf&gt; element that are not detailed in this figure.</em></p>
<p>The following attributes reside directly within the &lt;mbf&gt; element:</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 16%" />
<col style="width: 68%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Element</p></th>
<th class="head"><p>Attributes</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;mbf&gt;</p></td>
<td><p>All neuromorphological data is contained within
the &lt;mbf&gt; element.</p></td>
</tr>
<tr class="row-odd"><td rowspan="5"></td>
<td><p>version</p></td>
<td><p>Reports the version number of the
neuromorphological data file structure. It is
necessary to define the expected data structure
for the document that follows to ensure older
data versions are supported. The current
neuromorphological data file version number is
4.0.</p></td>
</tr>
<tr class="row-even"><td><p>appname</p></td>
<td><p>The name of the software that generated the
neuromorphological data file.</p></td>
</tr>
<tr class="row-odd"><td><p>appversion</p></td>
<td><p>The version of the software that generated the
neuromorphological data file. MBF Bioscience
software versions are reported with the year,
version, and minor version (YYYY.V.M).</p></td>
</tr>
<tr class="row-even"><td><p>apprrid</p></td>
<td><p>The Research Resource Identifier (RRID) for the
software application.</p></td>
</tr>
<tr class="row-odd"><td><p>insrrid</p></td>
<td><p>Institution Research Resource Identifier (RRID).</p></td>
</tr>
</tbody>
</table>
<p><em>Table 2 The &lt;mbf&gt; element, attributes, and descriptions for all fields.</em></p>
<p>The remaining portion of the header information is described in detail below. File header information includes necessary metadata that helps to describe the data included in the neuromorphological file.</p>
<div class="section" id="description">
<h3>4.1 &lt;description&gt;<a class="headerlink" href="#description" title="Permalink to this headline">¶</a></h3>
<p>The &lt;description&gt; element is a string of user optional text describing the contents of the data file. The format of the text is character data indicated by the CDATA section, used to differentiate the text block from markup.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;description&gt;</span><span class="cp">&lt;![CDATA[Example description of image segmentation file]]&gt;</span><span class="nt">&lt;/description&gt;</span>
</pre></div>
</div>
<p><em>Figure 5 The &lt;description&gt; element contains the CDATA start section &lt;![CDATA[ and CDATA end section ]] to indicate a large string of text that is not a data element.</em></p>
</div>
<div class="section" id="filefacts">
<h3>4.2 &lt;filefacts&gt;<a class="headerlink" href="#filefacts" title="Permalink to this headline">¶</a></h3>
<p>The &lt;filefacts&gt; element stores sequentially ordered serial Z sections for the data file that correspond to physically or virtually sectioned histologic tissue. If no sections have been created for a data file, the only child element of the &lt;filefacts&gt; is the &lt;sectionmanager&gt;. In this case, the values of the &lt;sectionmanager&gt; attributes indicate there have been no sections created and the child element should be disregarded.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;filefacts&gt;</span>
      <span class="nt">&lt;sectionmanager</span> <span class="na">currentsection=</span><span class="s">&quot;&quot;</span> <span class="na">sectioninterval=</span><span class="s">&quot;0&quot;</span> <span class="na">startingsection=</span><span class="s">&quot;0&quot;</span><span class="nt">/&gt;</span>
<span class="nt">&lt;/filefacts&gt;</span>
</pre></div>
</div>
<p><em>Figure 6 displays the &lt;filefacts&gt; of a data file with no sections. Note the current section has no value, the sectioninterval value is “0”, and the startingsection value is “0”.</em></p>
<p>If serial Z sections have been created for the data file, this child element contains meaningful parameters about the sections. Child &lt;section&gt; elements for each Z section are stored in the &lt;filefacts&gt; element.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;filefacts&gt;</span>
   <span class="nt">&lt;section</span> <span class="na">sid=</span><span class="s">&quot;S1&quot;</span> <span class="na">name=</span><span class="s">&quot;Section N0&quot;</span> <span class="na">top=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">cutthickness=</span><span class="s">&quot;n.nn&quot;</span> <span class="na">mountedthickness=</span><span class="s">&quot;n.nn&quot;</span><span class="nt">/&gt;</span>
   <span class="nt">&lt;section</span> <span class="na">sid=</span><span class="s">&quot;S2&quot;</span> <span class="na">name=</span><span class="s">&quot;Section N1&quot;</span> <span class="na">top=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">cutthickness=</span><span class="s">&quot;n.nn&quot;</span> <span class="na">mountedthickness=</span><span class="s">&quot;n.nn&quot;</span><span class="nt">/&gt;</span>
   <span class="nt">&lt;section</span> <span class="na">sid=</span><span class="s">&quot;S3&quot;</span> <span class="na">name=</span><span class="s">&quot;Section N2&quot;</span> <span class="na">top=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">cutthickness=</span><span class="s">&quot;n.nn&quot;</span> <span class="na">mountedthickness=</span><span class="s">&quot;n.nn&quot;</span><span class="nt">/&gt;</span>
   <span class="nt">&lt;sectionmanager</span> <span class="na">currentsection=</span><span class="s">&quot;Section Nx&quot;</span> <span class="na">sectioninterval=</span><span class="s">&quot;i&quot;</span> <span class="na">startingsection=</span><span class="s">&quot;N0&quot;</span><span class="nt">/&gt;</span>
<span class="nt">&lt;/filefacts&gt;</span>
</pre></div>
</div>
<p><em>Figure 7 The &lt;filefacts&gt; of a data file with serial sections. Each section in the data file has a corresponding &lt;section&gt; element. The attributes of the &lt;sectionmanager&gt; element are now significant values.</em></p>
<p>All child elements and attributes of the &lt;filefacts&gt; are defined in Table 3.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 15%" />
<col style="width: 23%" />
<col style="width: 63%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Element</p></th>
<th class="head"><p>Attributes</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;filefacts&gt;</p></td>
<td><p>Stores sequentially ordered serial Z sections
for the data file that corresponds physically or
virtually sectioned histologic tissue.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>&lt;section&gt;</p></td>
<td><p>Each serial Z section generated receives a
&lt;section&gt; element that includes information for
the unique section.</p></td>
</tr>
<tr class="row-even"><td rowspan="5"></td>
<td><p>sid</p></td>
<td><p>A sequential identifier for each section
(S1, S2, S3… SN).</p></td>
</tr>
<tr class="row-odd"><td><p>name</p></td>
<td><p>Default names for each section are “Section N”,
where N is incremented by the sectioninterval.
The names can be user modified.</p></td>
</tr>
<tr class="row-even"><td><p>top</p></td>
<td><p>The highest Z value (µm) of the serial section.</p></td>
</tr>
<tr class="row-odd"><td><p>cutthickness</p></td>
<td><p>Equal to the distance from one section to the
next (µm).</p></td>
</tr>
<tr class="row-even"><td><p>mountedthickness</p></td>
<td><p>Used to represent the physical thickness (µm) of
the tissue on the slide following section
processing and the associated shrinkage.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>&lt;sectionmanager&gt;</p></td>
<td><p>Contains parameters about the sections created
for a data file.</p></td>
</tr>
<tr class="row-even"><td rowspan="3"></td>
<td><p>currentsection</p></td>
<td><p>The section of the data file the user was on when
the document was last saved.</p></td>
</tr>
<tr class="row-odd"><td><p>sectioninterval</p></td>
<td><p>Indicates the interval of the section names. A
section interval i with a startingsection at N0
would produce section names: Section N0, Section
N0+i, Section N0+2i, Section N0+3i … Section
N0+(n)i where n is the total number of sections.</p></td>
</tr>
<tr class="row-even"><td><p>startingsection</p></td>
<td><p>The first section in the interval list.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 3 The &lt;filefacts&gt; element, its child elements, attributes, and descriptions for all fields.</em></p>
</div>
<div class="section" id="sparcdata">
<h3>4.3 &lt;sparcdata&gt;<a class="headerlink" href="#sparcdata" title="Permalink to this headline">¶</a></h3>
<p>The &lt;sparcdata&gt; element stores additional subject and annotation metadata. This information allows neuromorphological data files to be queried by species, subject ID, sex, age, and organ of the image sample origin.</p>
<p>MBF products interface with an external database, Scicrunch that maintains lists of anatomical terms separated by organ, species, and atlas/parcellation scheme. Each anatomy term is associated with a unique identifier and is approved by a team of anatomical experts. The &lt;sparcdata&gt; element informs the species, subject, sex, and age of the image sample segmented along with the exact anatomical list used to segment the image.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;sparcdata&gt;</span>
  <span class="nt">&lt;subject</span> <span class="na">species=</span><span class="s">&quot;http://purl.obolibrary.org/obo/NCBITaxon_XXXXXX&quot;</span> <span class="na">subjectid=</span><span class="s">&quot;SUBJECT_001&quot;</span> <span class="na">sex=</span><span class="s">&quot;Male&quot;</span> <span class="na">age=</span><span class="s">&quot;14 Weeks&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;atlas</span> <span class="na">organ=</span><span class="s">&quot;Brain&quot;</span> <span class="na">label=</span><span class="s">&quot;Allen Mouse Brain Atlas Terminology&quot;</span> <span class="na">rootid=</span><span class="s">&quot;http://purl.org/sig/ont/fma/fmaXXXX&quot;</span><span class="nt">/&gt;</span>
<span class="nt">&lt;/sparcdata&gt;</span>
</pre></div>
</div>
<p><em>Figure 8 A neuromorphological data file related to a microscopy image sample from a 14-week old male mouse brain delineated using anatomical terminology from the Allen Mouse Brain Atlas [3]. The species and atlas rootid provide URL links to the term lists origin.</em></p>
<p>The &lt;sparcdata&gt; child elements and their attributes are detailed in Table 4.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 16%" />
<col style="width: 16%" />
<col style="width: 68%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Element</p></th>
<th class="head"><p>Attributes</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;sparcdata&gt;</p></td>
<td><p>Additional subject and annotation metadata.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>&lt;subject&gt;</p></td>
<td><p>Metadata describing the subject from which the
image sample originated.</p></td>
</tr>
<tr class="row-even"><td rowspan="4"></td>
<td><p>species</p></td>
<td><p>The subject’s species is reported using the NCBI
Taxon identifier to ensure databases can query
this information and relate synonyms to the ID.</p></td>
</tr>
<tr class="row-odd"><td><p>subjectid</p></td>
<td><p>A user-defined subject identifier based on the
animal sample naming scheme that the researcher
prefers.</p></td>
</tr>
<tr class="row-even"><td><p>sex</p></td>
<td><p>Subject sex can either be reported as male or
female. Can be set to “blind to condition” while
data collection is in progress if this is part of
the experimental design.</p></td>
</tr>
<tr class="row-odd"><td><p>age</p></td>
<td><p>Age is a user-defined value and can be reported
in a unit of either days, weeks, months, or
years. Can be set to “blind to condition” while
data collection is in progress if this is part of
the experimental design.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>&lt;atlas&gt;</p></td>
<td><p>Defines the anatomy terminology list used to
annotate an image.</p></td>
</tr>
<tr class="row-odd"><td rowspan="3"></td>
<td><p>organ</p></td>
<td><p>The organ attribute indicates which subset of
terms was used to annotate, grouped by major
organ.</p></td>
</tr>
<tr class="row-even"><td><p>label</p></td>
<td><p>Indicates the parcellation scheme or atlas.</p></td>
</tr>
<tr class="row-odd"><td><p>rootid</p></td>
<td><p>A URL link to the term lists origin.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 4 The &lt;sparcdata&gt; element, its child elements, attributes, and descriptions for all fields.</em></p>
</div>
<div class="section" id="property-name-timepointmanager">
<h3>4.4 &lt;property name=”TimePointManager”&gt;<a class="headerlink" href="#property-name-timepointmanager" title="Permalink to this headline">¶</a></h3>
<p>The &lt;property&gt; element named “TimePointManager” will always be present in the neuromorphological data files, however, it is not used at this time. The element can be disregarded.</p>
</div>
<div class="section" id="images">
<h3>4.5 &lt;images&gt;<a class="headerlink" href="#images" title="Permalink to this headline">¶</a></h3>
<p>Raw image data is not saved within neuromorphological data files, rather they are linked with a file location and name and other information about the image.</p>
<p>The &lt;images&gt; element in the header can contain one or many &lt;image&gt; elements enabling the neuromorphological data file to be associated with any number of source images. The images can be either 2D (a single image plane) or 3D (multiple image planes from a single file or multiple files). Image data can be combined in several ways inside the neuromorphological data file. The simplest is the single 3D or 2D image file. See an example in Figure 9.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;images&gt;</span>
  <span class="nt">&lt;image&gt;</span>
    <span class="nt">&lt;filename&gt;</span>\\FilePath\ImageName<span class="nt">&lt;/filename&gt;&lt;channels</span> <span class="na">merge=</span><span class="s">&quot;no&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;channel</span> <span class="na">id=</span><span class="s">&quot;red&quot;</span> <span class="na">source=</span><span class="s">&quot;none&quot;</span><span class="nt">/&gt;</span>
      <span class="nt">&lt;channel</span> <span class="na">id=</span><span class="s">&quot;green&quot;</span> <span class="na">source=</span><span class="s">&quot;none&quot;</span><span class="nt">/&gt;</span>
      <span class="nt">&lt;channel</span> <span class="na">id=</span><span class="s">&quot;blue&quot;</span> <span class="na">source=</span><span class="s">&quot;none&quot;</span><span class="nt">/&gt;</span>
    <span class="nt">&lt;/channels&gt;</span>
    <span class="nt">&lt;scale</span> <span class="na">x=</span><span class="s">&quot;x.xxx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yyy&quot;</span><span class="nt">/&gt;</span>
    <span class="nt">&lt;coord</span> <span class="na">x=</span><span class="s">&quot;x.xxx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yyy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zzz&quot;</span><span class="nt">/&gt;</span>
    <span class="nt">&lt;zspacing</span> <span class="na">z=</span><span class="s">&quot;z.zzz&quot;</span> <span class="na">slices=</span><span class="s">&quot;n&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;/image&gt;</span>
<span class="nt">&lt;/images&gt;</span>
</pre></div>
</div>
<p><em>Figure 9 An &lt;images&gt; element with only one image associated with the data file.</em></p>
<p>Within each &lt;image&gt; element, five nested elements provide details pertaining to that image(s).</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 23%" />
<col style="width: 22%" />
<col style="width: 55%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;image&gt;</p></td>
<td><p>The &lt;images&gt; element can contain
one or many nested &lt;image&gt; child
elements. This enables the
neuromorphological data file to be
associated with any number of
source images.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>&lt;filename&gt;</p></td>
<td><p>The windows file path and name of
the associated image.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>&lt;channels&gt;</p></td>
<td><p>Defines the colors of each channel
if the channels are merged from
separate image files.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>merge</p></td>
<td><p>The value of the merge attribute
will be “yes” if the channels have
been merged from separate image
files and “no” if they have not.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>&lt;channel&gt;</p></td>
<td><p>Three &lt;channel&gt; child elements for
the color channels, red, green, and
blue are always reported. Their
unique id and source attributes
define the channel color and
source.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>id</p></td>
<td><p>Identifies the pseudocolor of the
color channel.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>source</p></td>
<td><p>Records the selected color channel
number from the corresponding
source image in the &lt;filename&gt;
element. Starts at 0.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>&lt;scale&gt;</p></td>
<td><p>The size of each pixel in x and y.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>x</p></td>
<td><p>The X scaling of the image in
µm/pixel.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>y</p></td>
<td><p>The Y scaling of the image in
µm/pixel.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>&lt;coord&gt;</p></td>
<td><p>Reports the attributes and values
for the X, Y, and Z coordinates of
the upper, left-hand corner of the
image.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>x</p></td>
<td><p>The X location of the image origin
(µm).</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>y</p></td>
<td><p>The Y location of the image origin
(µm).</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>z</p></td>
<td><p>The Z location of the image origin
(µm).</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>&lt;zspacing&gt;</p></td>
<td><p>The zspacing element uses the z and
slices attributes to describe the
relationship of image planes that
compile into a 3D image.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>z</p></td>
<td><p>Spacing between each image plane in
micrometer units.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>slices</p></td>
<td><p>The total number of image planes in
a 3D image.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 5 The &lt;image&gt; element, its child elements, attributes, and descriptions for all fields.</em></p>
<p>A neuromorphological data file with multiple, distinct images associated to the same file may include many &lt;image&gt; child elements in the &lt;images&gt; element. Each image will have its own set of child elements and attributes.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;images&gt;</span>
  <span class="nt">&lt;image&gt;</span>
    <span class="nt">&lt;filename&gt;</span>\\FilePath\Image[1]<span class="nt">&lt;/filename&gt;</span>
    <span class="nt">&lt;channels</span> <span class="na">merge=</span><span class="s">&quot;no&quot;</span><span class="nt">&gt;&lt;/channels&gt;</span>
    <span class="nt">&lt;scale/&gt;</span>
    <span class="nt">&lt;coord/&gt;</span>
    <span class="nt">&lt;zspacing/&gt;</span>
  <span class="nt">&lt;/image&gt;</span>
  <span class="nt">&lt;image&gt;</span>
    <span class="nt">&lt;filename&gt;</span>\\FilePath\Image[2]<span class="nt">&lt;/filename&gt;</span>
    <span class="nt">&lt;channels</span> <span class="na">merge=</span><span class="s">&quot;no&quot;</span><span class="nt">&gt;&lt;/channels&gt;</span>
    <span class="nt">&lt;scale/&gt;</span>
    <span class="nt">&lt;coord/&gt;</span>
    <span class="nt">&lt;zspacing/&gt;</span>
  <span class="nt">&lt;/image&gt;</span>
<span class="nt">&lt;/images&gt;</span>
</pre></div>
</div>
<p><em>Figure 10 An &lt;images&gt; element with multiple distinct images associated with the same data file. Note there are two separate &lt;image&gt; child elements within the &lt;images&gt; element. For concicity, the attributes of each child elements of each &lt;image&gt; have been excluded from the figure, but follow the same structure as see in Figure 9.</em></p>
<p>A 3D volume can be made up of many 2D image planes. These volumes are recorded within the neuromorphological data file with one &lt;image&gt; child element and an ordered list of &lt;filename&gt; children elements pointing to the location and file name for each image plane. Each 2D image plane is sequentially loaded from the list into a 3D image volume. The scale is consistent for every image plane. The &lt;coord&gt; reports the X, Y, and Z coordinates of the upper, left-hand corner of the first image plane. The distance between image planes and the number of slices is kept constant for the entire image volume.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;images&gt;</span>
      <span class="nt">&lt;image&gt;</span>
        <span class="nt">&lt;filename&gt;</span>FilePath\Image[1]<span class="nt">&lt;/filename&gt;</span>
            <span class="nt">&lt;filename&gt;</span>FilePath\Image[2]<span class="nt">&lt;/filename&gt;</span>
            <span class="nt">&lt;filename&gt;</span>FilePath\Image[3]<span class="nt">&lt;/filename&gt;</span>
            <span class="nt">&lt;filename&gt;</span>FilePath\Image[4]<span class="nt">&lt;/filename&gt;</span>
            ⋮
            <span class="nt">&lt;filename&gt;</span>FilePath\Image[N]<span class="nt">&lt;/filename&gt;</span>
            <span class="nt">&lt;channels</span> <span class="na">merge=</span><span class="s">&quot;no&quot;</span><span class="nt">&gt;&lt;/channels&gt;</span>
            <span class="nt">&lt;scale/&gt;</span>
            <span class="nt">&lt;coord/&gt;</span>
            <span class="nt">&lt;zspacing</span> <span class="na">z=</span><span class="s">&quot;z.zzz&quot;</span> <span class="na">slices=</span><span class="s">&quot;N&quot;</span><span class="nt">/&gt;</span>
      <span class="nt">&lt;/image&gt;</span>
    <span class="nt">&lt;/images&gt;</span>
</pre></div>
</div>
<p><em>Figure 11 A data file with many 2D image planes that make up a 3D image volume. The file path and image name for each 2D image are reported sequentially in the same &lt;image&gt; child element. In the &lt;zspacing&gt; element, the distance between each 2D image is reported, z=“z.zz”. The number of 2D image planes that construct the 3D volume will equal the total number of slices=”N”. Other image elements have been shortened to emphasize the important differences in this image organization. The &lt;channels&gt;, &lt;scale&gt;, and &lt;coord&gt; attributes are still included in these data files.</em></p>
<p>A multi-channel microscopy image can be generated by merging 2 or 3 source images that make up the image color channels. If the images are merged to create a multi-channel image, the &lt;channels&gt; merge attribute will have a value of “yes”. There will always be three &lt;filename&gt; elements that sequentially correspond to the &lt;channel&gt; elements (i.e. &lt;filename&gt;1 will always correspond to the red &lt;channel&gt; element). The &lt;channel&gt; id attribute reports the pseudocolor of the color channel (red, green, or blue). The source attribute defines the image channel number from the source image starting at 0. If no image is selected for a &lt;channel&gt; element, the &lt;filename&gt; will remain but no image path or name is reported and the &lt;channel&gt; source value will equal “none”.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;images&gt;</span>
      <span class="nt">&lt;image&gt;</span>
        <span class="nt">&lt;filename&gt;</span>FilePath\Image[1]<span class="nt">&lt;/filename&gt;</span>
        <span class="nt">&lt;filename&gt;</span>FilePath\Image[2]<span class="nt">&lt;/filename&gt;</span>
        <span class="nt">&lt;filename&gt;</span>FilePath\Image[1]<span class="nt">&lt;/filename&gt;</span>
        <span class="nt">&lt;channels</span> <span class="na">merge=</span><span class="s">&quot;yes&quot;</span><span class="nt">&gt;</span>
          <span class="nt">&lt;channel</span> <span class="na">id=</span><span class="s">&quot;red&quot;</span> <span class="na">source=</span><span class="s">&quot;0&quot;</span><span class="nt">/&gt;</span>
          <span class="nt">&lt;channel</span> <span class="na">id=</span><span class="s">&quot;green&quot;</span> <span class="na">source=</span><span class="s">&quot;2&quot;</span><span class="nt">/&gt;</span>
          <span class="nt">&lt;channel</span> <span class="na">id=</span><span class="s">&quot;blue&quot;</span> <span class="na">source=</span><span class="s">&quot;1&quot;</span><span class="nt">/&gt;</span>
        <span class="nt">&lt;/channels&gt;</span>
        <span class="nt">&lt;scale/&gt;</span>
        <span class="nt">&lt;coord/&gt;</span>
        <span class="nt">&lt;zspacing/&gt;</span>
      <span class="nt">&lt;/image&gt;</span>
    <span class="nt">&lt;/images&gt;</span>
</pre></div>
</div>
<p><em>Figure 12 A data file with multiple images merged into one multi-channel image. The first channel of Image[1], source=”0”, is displayed in MBF Bioscience software as red. The third channel of Image[2], source=”2”, is displayed in MBF Bioscience software as green. The second channel of Image[1], source”1”, is displayed in MBF Bioscience software as blue.</em></p>
</div>
<div class="section" id="thumbnail">
<h3>4.6 &lt;thumbnail&gt;<a class="headerlink" href="#thumbnail" title="Permalink to this headline">¶</a></h3>
<p>The &lt;thumbnail&gt; element stores data to create a small graphical (64x64) representation of the tracing data. Each of the 64 &lt;thumbnail-line&gt; elements represent one row in the thumbnail. The 64 pixels in each row are represented with a 3-byte hexadecimal (RRGGBB) alphanumeric character.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;thumbnail</span> <span class="na">cols=</span><span class="s">&quot;64&quot;</span> <span class="na">rows=</span><span class="s">&quot;64&quot;</span><span class="nt">&gt;</span>
   <span class="nt">&lt;thumbnail-line&gt;</span>0xRR1GG1BB1RR2GG2BB2...RR64GG64BB64<span class="nt">&lt;/thumbnail-line&gt;</span>
   <span class="nt">&lt;thumbnail-line&gt;&lt;/thumbnail-line&gt;</span>
    ⋮
<span class="nt">&lt;/thumbnail&gt;</span>
</pre></div>
</div>
<p><em>Figure 13 The &lt;thumbnail&gt; element, child elements, attributes, and values as they apear in the segmentation data file. The cols and rows attributes define the pixel dimentions of the 2D traced data thumbnail. A &lt;thumbnail-line&gt; child elements is present for every row of the &lt;thumbnail&gt;, but it is abbriviated here for concision. There are RRGGBB characters for each column of the data file. In the figure, the subscript numbers indicate the column number for the character.</em></p>
</div>
</div>
<div class="section" id="trace-data">
<h2>5.Trace Data<a class="headerlink" href="#trace-data" title="Permalink to this headline">¶</a></h2>
<p>The traced data elements include all data models of neuromorphological structures and additional image annotations. A data file will not necessarily contain all traced data elements. Typically a data file will include more than one of a single traced element in a data file. For example, there may be many contours of the same contour type and/or of a different contour type depending on what has been segmented. Figure 14 demonstrates the general organization of all trace data elements with a neuromorphological data file.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="cp">&lt;?xml version=&quot;1.0&quot; encoding=&quot;ISO-8859-1&quot;?&gt;</span>
<span class="nt">&lt;mbf&gt;</span>

  HEADER ELEMENTS

  <span class="nt">&lt;contour&gt;&lt;/contour&gt;</span>
  <span class="nt">&lt;marker&gt;&lt;/marker&gt;</span>
  <span class="nt">&lt;arrow&gt;&lt;/arrow&gt;</span>
  <span class="nt">&lt;tree&gt;</span>
    <span class="nt">&lt;spine&gt;&lt;/spine&gt;</span>
    <span class="nt">&lt;varicosity&gt;&lt;/varicosity&gt;</span>
  <span class="nt">&lt;/tree&gt;</span>
  <span class="nt">&lt;vessel&gt;&lt;/vessel&gt;</span>
  <span class="nt">&lt;text&gt;&lt;/text&gt;</span>
  <span class="nt">&lt;scalebar&gt;&lt;/scalebar&gt;</span>
<span class="nt">&lt;/mbf&gt;</span>
</pre></div>
</div>
<p><em>Figure 14 Demonstrates the organization of all traced data elements in the Neuromorphological file format. All traced data and header elements are child elements of  the &lt;mbf&gt; element. The &lt;spine&gt; and &lt;varicosity&gt; elements are child elements of the &lt;tree&gt;. The order and number of these elements within the parent &lt;tree&gt; may vary. The header elements are represented with the HEADER ELEMENTS comment and do not actually appear this way in the data file. They have been abbreviated to focus attention on the traced data elements.</em></p>
<div class="section" id="common-components">
<h3>5.1 Common Components<a class="headerlink" href="#common-components" title="Permalink to this headline">¶</a></h3>
<div class="section" id="color">
<h4>5.1.1 Color<a class="headerlink" href="#color" title="Permalink to this headline">¶</a></h4>
<p>The color attribute is seen in many neuromorphological file elements. It uses a 3-byte hexadecimal (RRGGBB) alphanumeric character to represent the color of traced structures.</p>
</div>
<div class="section" id="point">
<h4>5.1.2 &lt;point&gt;<a class="headerlink" href="#point" title="Permalink to this headline">¶</a></h4>
<p>The nested element &lt;point&gt; is included in several elements of the neuromorphological  file format to denote a point at a distinct X, Y, and Z location with a 3D micrometer coordinate space. The &lt;point&gt; element also includes a diameter value indicating the thickness at that point. These coordinates are represented using the corresponding x, y, z, and d attributes and are always in micrometer units.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
</pre></div>
</div>
<p><em>Figure 15 The &lt;point&gt; element, attributes, and values as they apear in the segmentation data file.</em></p>
</div>
<div class="section" id="property-name-channel">
<h4>5.1.3 &lt;property name=”Channel”&gt;<a class="headerlink" href="#property-name-channel" title="Permalink to this headline">¶</a></h4>
<p>The “Channel” &lt;property&gt; indicates the image color channel used to trace an object. If a detection is performed on only one of the image color channels, then that image channel is reported for the detected data element. If the detection is performed on two or more color channels, the &lt;property name= “Channel”&gt; element will not be written for that structure. If the image is monochrome, the &lt;property name= “Channel”&gt; element will not be written for any structure. Child elements are expected to be detected in the same color channel as their parent structure. For example, a &lt;branch&gt;, &lt;spine&gt;, or &lt;varicosity&gt; is expected to be detected in the same color channel as the associated &lt;tree&gt;.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span>Line#  Neuromorphological Data File
[0]    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Channel&quot;</span><span class="nt">&gt;</span>
[1]    <span class="nt">&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;</span>
[2]    <span class="nt">&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;</span>
[3]    <span class="nt">&lt;c&gt;</span>#RRGGBB<span class="nt">&lt;/c&gt;&lt;/property&gt;</span>
</pre></div>
</div>
<p><em>Figure 16 The &lt;property name=”Channel” &gt; element, attributes, and values. The line numbers and return spaces present in the figure above were added for clarity and do not exist in the data file structure. The line numbers correspond with the numbered values in Table 6. The “Channel” values actually appear as a string.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 26%" />
<col style="width: 20%" />
<col style="width: 26%" />
<col style="width: 28%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Line #</strong></p></th>
<th class="head"><p><strong>Value</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="3"><p>name=”Channel”</p></td>
<td><p>Contains three
values that
describe the
channel an
element was
detected in.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[1]</p></td>
<td><p>version</p></td>
<td><p>The Channel
child element’s
format version.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[2]</p></td>
<td><p>channel number</p></td>
<td><p>Indicates the
color channel
number of the
source image a
trace element
was detected in
starting at 0.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[3]</p></td>
<td><p>channel color</p></td>
<td><p>The color of
this channel is
reported as
a hex
adecimal value.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 6 The &lt;property name=”Channel”&gt; element, values, and descriptions for all fields.</em></p>
</div>
<div class="section" id="property-name-set">
<h4>5.1.4 &lt;property name=”Set”&gt;<a class="headerlink" href="#property-name-set" title="Permalink to this headline">¶</a></h4>
<p>The &lt;property&gt; named “Set” can be found in any trace data element. The &lt;property&gt; is used to name and group one or many trace data elements. These elements can either be the same type (ex. just &lt;tree&gt; elements) or different types (ex. &lt;marker&gt;, &lt;tree&gt;, and &lt;contour&gt; elements). A set has one value that is a text string indicating the name of that set. An element can be associated with multiple “Set” properties.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;contour</span> <span class="na">name=</span><span class="s">&quot;Soma 1&quot;</span> <span class="na">color=</span><span class="s">&quot;#FFFF00&quot;</span> <span class="na">closed=</span><span class="s">&quot;true&quot;</span> <span class="na">shape=</span><span class="s">&quot;Contour&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;GUID&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;FillDensity&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Set&quot;</span><span class="nt">&gt;&lt;s&gt;</span>EXAMPLE SET NAME<span class="nt">&lt;/s&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
<span class="nt">&lt;/contour&gt;</span>
</pre></div>
</div>
<p><em>Figure 17 An example of the  “Set”  &lt;property&gt; and its text string value, &lt;s&gt;EXAMPLE SET NAME&lt;/s&gt;. In this example, the &lt;contour&gt; element “Soma 1” has been placed into a set called EXAMPLE SET NAME. The &lt;contour&gt; includes the child  element &lt;property name=”Set”&gt; and the unique set name to indicate it belongs to the group of trace elements.</em></p>
</div>
</div>
<div class="section" id="marker">
<h3>5.2 &lt;marker&gt;<a class="headerlink" href="#marker" title="Permalink to this headline">¶</a></h3>
<p>The neuromorphological file format uses the &lt;marker&gt; element to represent single points in the data file. The only child element of the &lt;marker&gt; is the &lt;point&gt; element indicating the X, Y, and Z, location of the &lt;marker&gt; and the diameter of the point. Each &lt;point&gt; represents the center of one marker of specific marker type. A &lt;marker&gt; type can hold one or many &lt;point&gt; elements, each representing an independent marked location.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;marker</span> <span class="na">type=</span><span class="s">&quot;Plus&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">name=</span><span class="s">&quot;ExampleName&quot;</span> <span class="na">varicosity=</span><span class="s">&quot;false&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Channel&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
   ⋮
<span class="nt">&lt;/marker&gt;</span>
</pre></div>
</div>
<p><em>Figure 18 The &lt;marker&gt; element, child elements, attributes, and values as they apear in the segmentation data file. This &lt;marker&gt; type is Plus and is linked to a + symobol.</em></p>
<p>Each &lt;marker&gt; attribute, including type, color, name, and varicosity, is defined in Table 7.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;marker&gt;</p></td>
<td><p>A symbol placed in a
single point location
(x, y, and z).</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>type</p></td>
<td><p>A text field that
describes the symbol
shape used to mark a
point.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>color</p></td>
<td><p>Indicates the marker’s
symbol color,
represented with a
hexadecimal RGB value.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>name</p></td>
<td><p>A user-defined name.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>varicosity</p></td>
<td><p>This value will be set
to true if the marker is
used to mark a
varicosity, otherwise,
the value is set to
false to indicate a
typical marker element.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 7 The &lt;marker&gt; element, child elements, and all attributes. Each data component includes a corresponding description.</em></p>
<p>Makers can be associated with other traced objects including &lt;contour&gt; and &lt;tree&gt; elements. This section highlights the change in location of the &lt;marker&gt; when it is associated with another element. The &lt;marker&gt; becomes a child element to the parent &lt;contour&gt; or &lt;tree&gt; it is associated with and is always placed after the last &lt;point&gt; of the parent element.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;contour</span> <span class="na">name=</span><span class="s">&quot;ContourName&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">closed=</span><span class="s">&quot;true&quot;</span> <span class="na">shape=</span><span class="s">&quot;Contour&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;FillDensity&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;marker</span> <span class="na">type=</span><span class="s">&quot;Circle1&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRBBGG&quot;</span> <span class="na">name=</span><span class="s">&quot;ExampleName&quot;</span> <span class="na">varicosity=</span><span class="s">&quot;false&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;/marker&gt;</span>
<span class="nt">&lt;/contour&gt;</span>
</pre></div>
</div>
<p><em>Figure 19 A &lt;contour&gt; element with a child &lt;marker&gt; element representing an association of the two elements.</em></p>
</div>
<div class="section" id="punctum">
<h3>5.3 Punctum<a class="headerlink" href="#punctum" title="Permalink to this headline">¶</a></h3>
<p>Punctum volumes are represented with an extended version of the &lt;marker&gt; element. They contain all of the &lt;marker&gt; attributes, which are detailed in the &lt;marker&gt; section above. All punctum &lt;marker&gt; elements have the value “Punctum” for the name attribute differentiating them from a typical &lt;marker&gt; element. Similar to &lt;marker&gt; elements, punctum also include one &lt;point&gt; child element. This &lt;point&gt; describes the punctum’s center of mass, reporting an X, Y, and Z coordinate along with a diameter.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;marker</span> <span class="na">type=</span><span class="s">&quot;OpenCircle&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">name=</span><span class="s">&quot;Punctum&quot;</span> <span class="na">varicosity=</span><span class="s">&quot;false&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Channel&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Punctum&quot;</span><span class="nt">&gt;&lt;n&gt;&lt;/n&gt;&lt;n&gt;&lt;/n&gt;</span>...<span class="nt">&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;VolumeRLE&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
<span class="nt">&lt;/marker&gt;</span>
</pre></div>
</div>
<p><em>Figure 20 A “Punctum” represented with the &lt;marker&gt; element. In this figure, the &lt;property name=” “&gt; child elements exclude or abbrivate values for concision.</em></p>
<p>The punctum &lt;marker&gt; element also includes three &lt;property&gt; elements named “Channel”, “Punctum”, and “VolumeRLE”. The “Channel” is defined in the common components section of the document. “Punctum” and “VolumeRLE” are described in detail below.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span>Line#    Neuromorphological Data File
[0]      <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Punctum&quot;</span><span class="nt">&gt;</span>
[1]      <span class="nt">&lt;n&gt;</span>4<span class="nt">&lt;/n&gt;</span>
[2]      <span class="nt">&lt;n&gt;</span>n.nnnnn<span class="nt">&lt;/n&gt;</span>
[3]      <span class="nt">&lt;n&gt;</span>n.nnnnn<span class="nt">&lt;/n&gt;</span>
[4]      <span class="nt">&lt;n&gt;</span>n.nnnnn<span class="nt">&lt;/n&gt;</span>
[5]      <span class="nt">&lt;n&gt;</span>n<span class="nt">&lt;/n&gt;</span>
[6]      <span class="nt">&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;</span>
[7]      <span class="nt">&lt;n&gt;</span>n.nnnnn<span class="nt">&lt;/n&gt;</span>
[8]      <span class="nt">&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;</span>
[9]      <span class="nt">&lt;n&gt;</span>2<span class="nt">&lt;/n&gt;</span>
[10]     <span class="nt">&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;</span>
[11]     <span class="nt">&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
</pre></div>
</div>
<p><em>Figure 21 The “Punctum” &lt;property&gt; includes a string of 11 numbers. The line numbers and return spaces present in the figure above were added for clarity and do not exist in the data file structure. The line numbers correspond with the numbered values in Table 8. The “Punctum” values actually appear as a string as demonstrated in the “Punctum” &lt;property&gt; of Figure 20.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 26%" />
<col style="width: 19%" />
<col style="width: 27%" />
<col style="width: 27%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Line #</strong></p></th>
<th class="head"><p><strong>Value</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="3"><p>name=”Punctum”</p></td>
<td><p>A string of
numbers that
report the
puncta metrics.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[1]</p></td>
<td><p>version</p></td>
<td><p>The &lt;property
name=“Punctum”&gt;
format version.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[2]</p></td>
<td><p>spread</p></td>
<td><p>Distance of the
farthest voxel
from the center
of mass of the
punctum.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[3]</p></td>
<td><p>mean luminance</p></td>
<td><p>The average
luminance of
all the voxels
of the punctum.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[4]</p></td>
<td><p>surface area</p></td>
<td><p>The surface
area of the
mesh for the
detected
punctum.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[5]</p></td>
<td><p>voxel count</p></td>
<td><p>Total number of
voxels that
make up the
punctum.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[6]</p></td>
<td><p>2D flag</p></td>
<td><p>Equals 0 if the
punctum was
detected in 3D
and 1 if the
punctum was
detected in 2D.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[7]</p></td>
<td><p>volume</p></td>
<td><p>Volume of the
puncta in cubic
micrometers.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[8]</p></td>
<td><p>type</p></td>
<td><p>Reserved
integer value.
It may be used
as a custom
punctum type
modifier.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[9]</p></td>
<td><p>location</p></td>
<td><p>Indicates
punctum’s
colocalization
with
user-specified
structure:
0=colocalized,
1=proximal,
2=distal.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[10]</p></td>
<td><p>colocalized
fraction</p></td>
<td><p>A portion of
the volume that
is colocalized
with a
user-specified
structure.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[11]</p></td>
<td><p>proximal
fraction</p></td>
<td><p>A portion of
the volume that
is proximal to
a
user-specified
structure.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 8 The &lt;property name=”Punctum”&gt; element, its values, and a descriptions for all fields. The Line # column corresponds to the line numbers in Figure 21.</em></p>
<p>The final &lt;property&gt;, “VolumeRLE”, defines a cuboid enclosing the detected punctum volume. The “VolumeRLE” values are detailed in Table 9.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;VolumeRLE&quot;</span><span class="nt">&gt;&lt;s&gt;</span>x.xx y.yy z.zz VT Vx Vy Vz x.x y.y z.z B1 F1 B2 F2 ... Bn Fn<span class="nt">&lt;/s&gt;&lt;/property&gt;</span>
</pre></div>
</div>
<p><em>Figure 22 The “VolumeRLE” &lt;property&gt; includes a string of numbers that describe the voxels of the punctum’s detection cuboid. The variables in the figure correspond with one number, each defined in Table 9. The first two pairs of background and foreground values (B1, F1, B2, F2) are listed. The remaining background and foreground values are abbreviated in the figure for concision using the ellipsis. The final pair of background and foreground values (Bn, Fn) are the last values in the string.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 23%" />
<col style="width: 20%" />
<col style="width: 15%" />
<col style="width: 42%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Figure</strong>
<strong>Variable</strong></p></th>
<th class="head"><p><strong>Value</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="3"><p>name=”VolumeRLE”</p></td>
<td><p>A list of the foreground and
background voxels that
construct a cuboid enclosing
the punctum volume. The
foreground voxels are the
voxels within the cuboid
with a high luminance that
construct the punctum. In
contrast, background voxels
have a low luminance value
and are excluded from the
punctum volume. The first ten
values of the child element
describe the detection
cuboid. The rest of the
numbers are a series of pairs
reporting the background and
foreground voxels run lengths
within the cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>x.xx</p></td>
<td><p>x scaling</p></td>
<td><p>X scaling of the image the
punctum was detected from.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>y.yy</p></td>
<td><p>y scaling</p></td>
<td><p>Y scaling of the image the
punctum was detected from.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>z.zz</p></td>
<td><p>z scaling</p></td>
<td><p>Z scaling of the image the
punctum was detected from.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>V<sub>T</sub></p></td>
<td><p>voxel</p></td>
<td><p>Total number of foreground
voxels.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>V<sub>x</sub></p></td>
<td><p>x voxel
count</p></td>
<td><p>Total number of voxels in the
X direction that constructs
the detection cuboid.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>V<sub>y</sub></p></td>
<td><p>y voxel
count</p></td>
<td><p>Total number of voxels in the
Y direction that constructs
the detection cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>V<sub>y</sub></p></td>
<td><p>y voxel
count</p></td>
<td><p>Total number of voxels in the
Z direction that constructs
the detection cuboid.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>x.x</p></td>
<td><p>x origin</p></td>
<td><p>X origin of the cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>y.y</p></td>
<td><p>y origin</p></td>
<td><p>Y origin of the cuboid.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>z.z</p></td>
<td><p>z origin</p></td>
<td><p>Z origin of the cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>B<sub>1</sub></p></td>
<td><p>background
voxel
-run1</p></td>
<td><p>The number of background
voxels beginning at the
cuboids defined origin and
ending at the first
foreground voxel.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>F<sub>1</sub></p></td>
<td><p>foreground
voxel
-run1</p></td>
<td><p>The number of foreground
voxels between the last
reported background voxel and
the next background voxel.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 9 The &lt;property name=”VolumeRLE”&gt; element, its values, and a descriptions for all fields. The Figure Variable column corresponds to the variables in Figure 22.</em></p>
<p>The final values of the “VolumeRLE” &lt;property&gt; include a list of the foreground and background voxels that construct a cuboid enclosing the punctum volume. The foreground voxels are the voxels within the cuboid with a high luminance that make up the punctum’s volume. The foreground and background voxels are reported as a series of numbers indicating the voxels run lengths within the cuboid. The first run will count the number of background voxels beginning at the cuboids defined origin and ending at the first foreground voxel. The next number of the string reports the run of foreground voxels until the next background voxel is reached. The run-length moves positively in X until the cuboid boundary is reached. Next, the voxels are reported for the second line of Y until the first Z plane has been reported. The voxels are reported in the same pattern until all Z planes making up the cuboid have been reported.</p>
</div>
<div class="section" id="contour">
<h3>5.4 &lt;contour&gt;<a class="headerlink" href="#contour" title="Permalink to this headline">¶</a></h3>
<p>A &lt;contour&gt; element is a named list of sequentially connected points. The &lt;point&gt; child element is described above.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;contour</span> <span class="na">name=</span><span class="s">&quot;ContourName&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">closed=</span><span class="s">&quot;true&quot;</span> <span class="na">shape=</span><span class="s">&quot;Contour&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;GUID&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;FillDensity&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Channel&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;resolution&gt;</span>n.nnnnnn<span class="nt">&lt;/resolution&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;TraceAssociation&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
   ⋮
<span class="nt">&lt;/contour&gt;</span>
</pre></div>
</div>
<p><em>Figure 23 The &lt;contour&gt; element, child elements, attributes, and values as they apear in the segmentation data file. This contour is a closed contour indicating the first and last &lt;point&gt; elements are connected. In this figure, the &lt;property name=” “&gt; child elements exclude all values for concision. The &lt;point&gt; elements have been abbriviated in this figure. A contour usually contains a list of many &lt;point&gt; elements, connected in the order they are listed in the &lt;contour&gt;.</em></p>
<p>The &lt;contour&gt; element contains four attributes. Each is defined in Table 10.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;contour&gt;</p></td>
<td><p>A named list of
sequentially connected
points.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>name</p></td>
<td><p>The contour name.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>color</p></td>
<td><p>A hexadecimal RGB value
denotes the contour
color.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>closed</p></td>
<td><p>Indicates if the contour
is open or closed. When
the value is “true”, the
contour is closed,
meaning the start and
endpoint are connected.
The value “false”
indicates the contour is
an open and the start
and endpoints of the
contour are not
connected.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>shape</p></td>
<td><p>The value can be
“Contour”, “Circle” or
“Box”. If the shape
value is equal to
”Contour” then the
series of sequentially
connected points inform
the contour’s perimeter.
Contours with a “Circle”
value will always have 2
&lt;point&gt; child elements
defining the center
point and radius. A
contour with shape=“Box”
will also always have
two &lt;point&gt; child
elements indicating the
upper left and bottom
right points of a
rectangle.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 10 The &lt;contour&gt; element and its attributes. Each data component includes a corresponding description.</em></p>
<p>There are four &lt;property&gt; child elements that can be included in the &lt;contour&gt; element. They are named “GUID”, “FillDensity”, “Channel” and “TraceAssociation”.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;GUID&quot;</span><span class="nt">&gt;&lt;s&gt;&lt;/s&gt;&lt;/property&gt;</span>
<span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;FillDensity&quot;</span><span class="nt">&gt;&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
<span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Channel&quot;</span><span class="nt">&gt;&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;&lt;c&gt;</span>#RRGGBB<span class="nt">&lt;/c&gt;&lt;/property&gt;</span>
<span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;TraceAssociation&quot;</span><span class="nt">&gt;&lt;s&gt;</span>http://<span class="nt">&lt;/s&gt;&lt;/property&gt;</span>
</pre></div>
</div>
<p><em>Figure 24 Each &lt;property&gt; element of the contour with examples values. This example shows a contour with a “FillDensity” of 0, indicating the contour is not filled. The “Channel” property indicates the contour was detected on the first channel of the image which has a color reported as a hexadecimal of #RRGGBB. The TraceAssociation field includes a placeholder URL that links anatomy terms with a unique identifier.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 37%" />
<col style="width: 63%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>name=”GUID”</p></td>
<td><p>A unique contour identification number.</p></td>
</tr>
<tr class="row-odd"><td><p>name=”FillDensity”</p></td>
<td><p>Provides a number relating to the
contour’s fill percentage. 0 indicates
the contour has no fill and the polygon
is hollow. The value can equal a number
between 0 and 255. 100% fill density will
report a value of 255.</p></td>
</tr>
<tr class="row-even"><td><p>name=”Channel”</p></td>
<td><p>Contains three values that describe the
channel an element was detected in.
Detailed above.</p></td>
</tr>
<tr class="row-odd"><td><p>name=”TraceAssociation”</p></td>
<td><p>Lists a text string URL that is linked to
the anatomy term used for a contour and
provides a unique identifier for the
term.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 11 The &lt;property&gt; child elements of a &lt;contour&gt; and a description for each.</em></p>
<p>Contours are often used to delineate anatomical regions within image data. The “TraceAssociation” &lt;property&gt; of the &lt;contour&gt; stores a URL that is linked to the anatomy term used for a contour and includes the unique identifier for that term.</p>
<p>The final elements of the contour are &lt;resolution&gt; and the list of &lt;point&gt; elements. The &lt;resolution&gt; element contains a value equal to the size of the pixel as the contour was traced. The list of &lt;point&gt; elements are connected describing the contour.</p>
</div>
<div class="section" id="neurons">
<h3>5.5 Neurons<a class="headerlink" href="#neurons" title="Permalink to this headline">¶</a></h3>
<div class="section" id="tree">
<h4>5.5.1 &lt;tree&gt;<a class="headerlink" href="#tree" title="Permalink to this headline">¶</a></h4>
<p>In the neuromorphological file structure, the &lt;tree&gt; element is used to represent non-looping branching structures within microscopy images such as axons, dendrites, and airways.</p>
<a class="reference internal image-reference" href="_images/2.png"><img alt="" class="align-center" src="_images/2.png" style="width: 1000px; height: 450px;" /></a>
<p><em>Figure 25 (a) A condensed &lt;tree&gt; element of the Neuromorphological file format. All properties of &lt;tree&gt; and &lt;branch&gt; are fully depicted, however, the &lt;point&gt; elements have been abbreviated using an ellipsis to draw focus to the structure of the &lt;tree&gt;. The column on the left of the figure includes the name of the segment for the corresponding like of the data file. Each node, origin, and ending point is labeled with a gray arrow pointing to the appropriate line of the data file. (b) A diagram demonstrating the structure of a &lt;tree&gt; with each segment shown as a gray line and labeled with the segment name (ex. S2-2-2).  The origin (O), nodes (N) and endings (E) of the tree are marked with a circle. The root segment (S) begins with the origin (O) point and terminates with the node (N0). The child segments of N0, S1 and S2, terminate with nodes N1 and N2. The child segments of N1, S1-2 and S1-1 terminate with endings E1-1 and E1-2. N2 has two child segments, S2-1 and S2-2. Segment S2-1 has no bifurcations, so it terminates with ending E2-1. Segment S2-2 bifurcates at node N2-1. Lastly, the branches S2-2-1 and S2-2-2 terminate with endings E2-2-1 and E2-2-2.</em></p>
<p>Trees consist of an origin, branches, nodes, and endings. The starting point of &lt;tree&gt; element is referred to as the origin and the points that follow make up the root branch of the &lt;tree&gt;. All trees must have at least an origin and root branch, but typically have branching points called nodes. Nodes are where a segment of the &lt;tree&gt; splits into multiple &lt;branch&gt; child elements. The &lt;branch&gt; elements are made up of an ordered list of points that connect nodes to nodes, and nodes to endings. Endings are the last point of a &lt;branch&gt; or &lt;tree&gt; where the segment terminates.</p>
<p>Table 12 describes the attributes of a tree and additional child elements.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 23%" />
<col style="width: 57%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;tree&gt;</p></td>
<td><p>Used to model branching structures
with no loops. A tree is represented
with an ordered list of connected
points.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>color</p></td>
<td><p>Indicates the tree color via a
hexadecimal RGB value.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>type</p></td>
<td><p>A text field that A text field that
describes the tree type. Value
options currently include “Axon”,
“Dendrite”, and “Apical Dendrite”.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>leaf</p></td>
<td><p>Indicates the ending type for a tree
or branch. The value can equal
“Normal”, “High”, “Low”,
“Incomplete”, “Origin”, “Generated”
or “Midpoint” to describe how the
branch ends.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>&lt;zsmear&gt;</p></td>
<td><p>Used to alter the Used to alter the
displayed thickness of dendrites to
resemble the optical aberration in
z. This can be caused by both the
point spread function and by
refractive index mismatch between
the specimen and the lens immersion
medium. The diameter of a branch in
z is adjusted using the following
equation, Dz= Dxy*S, where Dxy is
the recorded centerline diameter on
the xy plane and S is the smear
factor. The smear factor is
calculated using this equation, S=α*
Dxyβ. The minimum diameter is 1.0
µm, even if S values are less than
1.0.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>alpha</p></td>
<td><p>Alpha (α) parameter in zsmear
equation.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>beta</p></td>
<td><p>Beta (β) parameter in zsmear
equation.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 12 The &lt;tree&gt; element, its child elements, and all attributes. Each data component includes a corresponding description.</em></p>
<p>There are two more child elements of the &lt;tree&gt;. Both varicosities and spines are related to a specific point location on a neuron. Therefore, it is necessary to represent this relationship in the neuromorphological data structure. The &lt;varicosity&gt; and &lt;spine&gt; elements are nested within their corresponding &lt;tree&gt; or &lt;branch&gt;, and are located between the &lt;tree&gt; or &lt;branch&gt; points that they were detected. Each of these elements is detailed in the corresponding sections below.</p>
<div class="section" id="spine">
<h5>5.5.1.1 &lt;spine&gt;<a class="headerlink" href="#spine" title="Permalink to this headline">¶</a></h5>
<p>Spines are small projections off of neuronal trees. They have a high postsynaptic density that helps to transmit electrical signals to the cell body . Because spines occur at a unique point location on a neuronal branch, the neuromorphological data format embeds the spine element in the associated branch, recorded between the tree points that the spine occurs. Each &lt;spine&gt; has a set of attributes, &lt;property&gt; elements, and one &lt;point&gt; element used to define the detected object. The &lt;point&gt; child element describes the spine head’s center of mass, reporting the X, Y, and Z coordinates along with the diameter of the spine head (see the last child element of the spine in Figure 26).</p>
<p>The neuromorphological file format includes two modeling methods for a spine, one more simple and succinct than the other. The simple modeling method is referred to as a ball and stick representation. When a spine is manually placed, the head location and thickness is defined using the spine &lt;point&gt; property. The &lt;point&gt; element of the parent &lt;tree&gt; of the &lt;spine&gt; defines the location that the spine connects to the branch.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;tree</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">type=</span><span class="s">&quot;Dendrite&quot;</span> <span class="na">leaf=</span><span class="s">&quot;Normal&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;xxx.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;yyy.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;zzz.zz&quot;</span> <span class="na">d=</span><span class="s">”ddd.dd&quot;/</span><span class="nt">&gt;</span>
  <span class="nt">&lt;spine</span> <span class="na">version=</span><span class="s">&quot;4&quot;</span> <span class="na">classification=</span><span class="s">&quot;detached&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Class&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Color&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Generated&quot;</span><span class="nt">&gt;&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;xxx.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;yyy.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;zzz.zz&quot;</span> <span class="na">d=</span><span class="s">”ddd.dd&quot;/</span><span class="nt">&gt;</span>
  <span class="nt">&lt;/spine&gt;</span>
  ⋮
<span class="nt">&lt;/tree&gt;</span>
</pre></div>
</div>
<p><em>Figure 26 A manually detected &lt;spine&gt; element nested with a &lt;tree&gt;. The values of all &lt;property&gt; child elements have been excluded to highlight the full structure of the &lt;spine&gt; element with the exception of the &lt;property name=”generated”&gt;. When this value is set to 0, the spine was manually detected.</em></p>
<p>Modeling the entire 3D structure of the &lt;spine&gt; is more complex, but provides exhaustive morphometric data for the &lt;spine&gt;.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;tree</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">type=</span><span class="s">&quot;Dendrite&quot;</span> <span class="na">leaf=</span><span class="s">&quot;Normal&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;spine</span> <span class="na">version=</span><span class="s">&quot;4&quot;</span> <span class="na">classification=</span><span class="s">&quot;stubby&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Class&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Color&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Volume&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Generated&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;GeneratedMetrics&quot;</span><span class="nt">&gt;&lt;n&gt;&lt;/n&gt;&lt;n&gt;</span>...<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Backbone&quot;</span><span class="nt">&gt;&lt;n&gt;&lt;/n&gt;&lt;n&gt;</span>...<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;VolumeRLE&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;xxx.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;yyy.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;zzz.zz&quot;</span> <span class="na">d=</span><span class="s">”ddd.dd&quot;/</span><span class="nt">&gt;</span>
  <span class="nt">&lt;/spine&gt;</span>
  ⋮
<span class="nt">&lt;/tree&gt;</span>
</pre></div>
</div>
<p><em>Figure 27 An automatically detected &lt;spine&gt; element nested with a &lt;tree&gt;. The values of all &lt;property&gt; child elements have been excluded to highlight the full structure of the &lt;spine&gt; element.</em></p>
<p>The attributes of the spine element are described in the section below.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 23%" />
<col style="width: 57%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;spine&gt;</p></td>
<td><p>The spine element is a child element
of the associated branch and is
nested within the tree points that
the spine occurs.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>version</p></td>
<td><p>Defines the data format version for
the spine child element.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>classification</p></td>
<td><p>Records the type of spine: none,
filopodium, mushroom, stubby, thin,
branched, detached, and other.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 13 The &lt;spine&gt; element and its attributes. Each data component includes a corresponding description.</em></p>
<p>Each spine element has 7 &lt;property&gt; elements named: “Class”, “Color”, “Volume”, “Generated”, “GeneratedMetrics”, “Backbone”, and “VolumeRLE”. Each &lt;property&gt; element and its values are detailed below.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Class&quot;</span><span class="nt">&gt;&lt;n&gt;</span>4<span class="nt">&lt;/n&gt;&lt;s&gt;</span>stubby<span class="nt">&lt;/s&gt;&lt;/property&gt;</span>
<span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Color&quot;</span><span class="nt">&gt;&lt;c&gt;</span>#RRGGBB<span class="nt">&lt;/c&gt;&lt;/property&gt;</span>
<span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Volume&quot;</span><span class="nt">&gt;&lt;n&gt;</span>n.nnnnn<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
<span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Generated&quot;</span><span class="nt">&gt;&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
</pre></div>
</div>
<p><em>Figure 28 (a) The first four &lt;property&gt; elements of a spine. The “Class” &lt;property&gt; indicates this is a stubby spine. The “Color” &lt;property&gt; of a spine includes an example of hexadecimal value for the spine color. The “Volume” property indicates a 6-digit real number of the calculated spine volume is reported in cubic micrometers units. The value &lt;n&gt;1&lt;/n&gt; reported in the “Generated” property indicates this spine as detected with an automated algorithm.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 27%" />
<col style="width: 24%" />
<col style="width: 48%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Value</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>name=”Class”</p></td>
<td><p>Defines the spine
classification.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>version</p></td>
<td><p>Defines the data format
version for the spine child
element.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>classification</p></td>
<td><p>Records the type of spine:
none, filopodium, mushroom,
stubby, thin, branched,
detached, and other.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>name=”Color”</p></td>
<td><p>Value defines the spine color.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>color</p></td>
<td><p>A hexadecimal RGB value.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>name=”Volume”</p></td>
<td><p>Value details the spine
volume.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>volume</p></td>
<td><p>Total spine volume reported in
cubic micrometers.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>name=”Generated”</p></td>
<td><p>Denotes when a spine is
detected by an automated
algorithm versus a manual
delineation.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>generated</p></td>
<td><p>If a spine is detected by an
automated algorithm, the value
will be 1. If a manual
algorithm was used, the value
is 0.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 14 The &lt;property&gt; child elements of a &lt;spine&gt;, their values, and a description for each field.</em></p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span>Line#   Neuromorphological Data File
 [0]    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;GeneratedMetrics&quot;</span><span class="nt">&gt;</span>
 [1]    <span class="nt">&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;</span>
 [2]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
 [3]    <span class="nt">&lt;n&gt;</span>d.dddddd<span class="nt">&lt;/n&gt;</span>
 [4]    <span class="nt">&lt;n&gt;</span>x.xxxxxx<span class="nt">&lt;/n&gt;</span>
 [5]    <span class="nt">&lt;n&gt;</span>y.yyyyyy<span class="nt">&lt;/n&gt;</span>
 [6]    <span class="nt">&lt;n&gt;</span>z.zzzzzz<span class="nt">&lt;/n&gt;</span>
 [7]    <span class="nt">&lt;n&gt;</span>d.dddddd<span class="nt">&lt;/n&gt;</span>
 [8]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
 [9]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
[10]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
[11]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
[12]    <span class="nt">&lt;n&gt;</span>n<span class="nt">&lt;/n&gt;</span>
[13]    <span class="nt">&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;</span>
[14]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
[15]    <span class="nt">&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;</span>
[16]    <span class="nt">&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;</span>
[17]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
[18]    <span class="nt">&lt;n&gt;</span>0<span class="nt">&lt;/n&gt;</span>
[19]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;</span>
[20]    <span class="nt">&lt;n&gt;</span>1<span class="nt">&lt;/n&gt;</span>
[21]    <span class="nt">&lt;n&gt;</span>n.nnnnnn<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>
</pre></div>
</div>
<p><em>Figure 29 The “GeneratedMetrics” &lt;property&gt; of a spine includes a string of 21 numbers. The line numbers and return spaces present in the figure above were added for clarity and do not exist in the data file structure. The line numbers correspond with the numbered values in Table 15. The “GeneratedMetrics” values actually appear as a string as demonstrated in the “GeneratedMetrics” &lt;property&gt; of Figure 27.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 23%" />
<col style="width: 11%" />
<col style="width: 15%" />
<col style="width: 52%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Line</strong>
<strong>#</strong></p></th>
<th class="head"><p><strong>Value</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="3"><p>name= “GeneratedMetrics”</p></td>
<td><p>A string of 21 numbers that report the
spine metrics.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[1]</p></td>
<td><p>Version</p></td>
<td><p>The data format version for the
GeneratedMetrics child element.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[2]</p></td>
<td><p>Total
Extent</p></td>
<td><p>Shortest distance from the furthest
identified voxel to the surface of the
dendrite.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[3]</p></td>
<td><p>Head Layer</p></td>
<td><p>The diameter of the spine head
calculated in the XY plane at the
center of the head layer.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[4]</p></td>
<td><p>Head Layer
Position X</p></td>
<td><p>X coordinate of the center of the head
of the spine.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[5]</p></td>
<td><p>Head Layer
Position Y</p></td>
<td><p>Y coordinate of the center of the head
of the spine.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[6]</p></td>
<td><p>Head Layer</p></td>
<td><p>Z coordinate of the center of the head
of the spine.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[7]</p></td>
<td><p>Neck Layer
Diameter</p></td>
<td><p>The diameter of the spine neck
calculated in the XY plane at the
center of the neck layer.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[8]</p></td>
<td><p>Neck Layer
Extent</p></td>
<td><p>Shortest distance from the center of
the neck layer diameter to the surface
of the dendrite.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[9]</p></td>
<td><p>Head Layer
Extent</p></td>
<td><p>Shortest distance from the center of
the head layer diameter to the surface
of the dendrite.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[10]</p></td>
<td><p>Surface
Area</p></td>
<td><p>The surface area of the mesh for the
detected spine.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[11]</p></td>
<td><p>Contact
Area</p></td>
<td><p>Cross-sectional area of contact
between the spine and its dendrite.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[12]</p></td>
<td><p>Voxel
Count</p></td>
<td><p>Total number of foreground voxels that
make up the spine.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[13]</p></td>
<td><p>Attached
Flag</p></td>
<td><p>0=spines is detached from the
dendrite, 1=spine is attached to the
dendrite.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[14]</p></td>
<td><p>Anchor
Radius</p></td>
<td><p>Refers to the radius of the dendrite
at the point where the spine attaches.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[15]</p></td>
<td><p>Anchor
Offset</p></td>
<td><p>Defines the position along the tree,
between the attachment and the
previous point that gives the precise
location of the center of the spine.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[16]</p></td>
<td><p>Auto
Classified
Flag</p></td>
<td><p>The value is equal to 1 if the spine
was automatically classified by an
algorithm and is equal to 0 if the
spine is manually classified.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[17]</p></td>
<td><p>Plane
Angle</p></td>
<td><p>Angle between the spine attachment
vector and the optical plane. Zero
means that the spine axis is parallel
to the optical plane. 90 (+/-) means
that the axis is perpendicular to the
optical plane.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[18]</p></td>
<td><p>Is 2D
Spine</p></td>
<td><p>0 if the spine was detected in 3D
image and 1 if the spine was detected
in 2D image.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[19]</p></td>
<td><p>Backbone
Length</p></td>
<td><p>Backbone length to the center minus
distance between dendritic surface and
insertion point on the centerline.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[20]</p></td>
<td><p>Classifier</p></td>
<td><p>0 = rule-based classifier using the
total extent for the length, or
1=rule-based classifier using the
spine backbone to compute the length.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[21]</p></td>
<td><p>Mean
Luminance</p></td>
<td><p>The mean luminance value of all
foreground voxels within the spine.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 15 The &lt;property name=”GeneratedMetrics”&gt; element, its values, and a descriptions for all fields. The Line # column corresponds to the line numbers in Figure 29.</em></p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span>Line#   Neuromorphological Data File
 [0]    <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Backbone&quot;</span><span class="nt">&gt;</span>
 [1]    <span class="nt">&lt;n&gt;</span>5<span class="nt">&lt;/n&gt;</span>
 [2]    <span class="nt">&lt;n&gt;</span>x.xxxxxx<span class="nt">&lt;/n&gt;</span>
 [3]    <span class="nt">&lt;n&gt;</span>y.yyyyyy<span class="nt">&lt;/n&gt;</span>
 [4]    <span class="nt">&lt;n&gt;</span>z.zzzzzz<span class="nt">&lt;/n&gt;</span>
 [5]    <span class="nt">&lt;n&gt;</span>d.dddddd<span class="nt">&lt;/n&gt;</span>
 [6]    <span class="nt">&lt;n&gt;</span>x.xxxxxx<span class="nt">&lt;/n&gt;</span>
 [7]    <span class="nt">&lt;n&gt;</span>y.yyyyyy<span class="nt">&lt;/n&gt;</span>
 [8]    <span class="nt">&lt;n&gt;</span>z.zzzzzz<span class="nt">&lt;/n&gt;</span>
 [9]    <span class="nt">&lt;n&gt;</span>d.dddddd<span class="nt">&lt;/n&gt;</span>
[10]    <span class="nt">&lt;n&gt;</span>x.xxxxxx<span class="nt">&lt;/n&gt;</span>
[11]    <span class="nt">&lt;n&gt;</span>y.yyyyyy<span class="nt">&lt;/n&gt;</span>
[12]    <span class="nt">&lt;n&gt;</span>z.zzzzzz<span class="nt">&lt;/n&gt;</span>
[13]    <span class="nt">&lt;n&gt;</span>d.dddddd<span class="nt">&lt;/n&gt;</span>
[14]    <span class="nt">&lt;n&gt;</span>x.xxxxxx<span class="nt">&lt;/n&gt;</span>
[15]    <span class="nt">&lt;n&gt;</span>y.yyyyyy<span class="nt">&lt;/n&gt;</span>
[16]    <span class="nt">&lt;n&gt;</span>z.zzzzzz<span class="nt">&lt;/n&gt;</span>
[17]    <span class="nt">&lt;n&gt;</span>d.dddddd<span class="nt">&lt;/n&gt;</span>
[18]    <span class="nt">&lt;n&gt;</span>x.xxxxxx<span class="nt">&lt;/n&gt;</span>
[19]    <span class="nt">&lt;n&gt;</span>y.yyyyyy<span class="nt">&lt;/n&gt;</span>
[20]    <span class="nt">&lt;n&gt;</span>z.zzzzzz<span class="nt">&lt;/n&gt;</span>
[21]    <span class="nt">&lt;n&gt;</span>d.dddddd<span class="nt">&lt;/n&gt;&lt;/property&gt;</span>                                                             A
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/3.png"><img alt="" class="align-center" src="_images/3.png" style="width: 300px; height: 275px;" /></a>
<p><em>Figure 30 (a) The “Backbone” &lt;property&gt; of a &lt;spine&gt; includes a string of numbers. The line numbers and return spaces present in the figure above were added for clarity and do not exist in the data file structure. The “Backbone” values actually appear as a string as demonstrated in the “Backbone” &lt;property&gt; of Figure 27. The first five line numbers correspond with the numbered values in Table 16 that provides a definition for each value. The values from line [2] through [21] make up each of the five &lt;spine&gt; coordinates (x, y, z, and d). The first point (x, y, z, and d) listed is the insertion point where the &lt;spine&gt; is located along the &lt;tree&gt;. (b) A diagram of a &lt;spine&gt; along a neuronal tree. The five points of the &lt;spine&gt; are represented with circles. The coordinates of these points are reported in the &lt;property name=”Backbone” number string including an x, y, and z location along with a thickness, d. The &lt;spine&gt; head is marked in the figure with a gray circle.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 27%" />
<col style="width: 13%" />
<col style="width: 15%" />
<col style="width: 45%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Line</strong>
<strong>#</strong></p></th>
<th class="head"><p><strong>Value</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="3"><p>name=”Backbone”</p></td>
<td><p>Describes the connected
points that make up the
reconstructed spine. The
points are listed in order
of proximity to the branch
the spine falls on.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[1]</p></td>
<td><p>Total
Points</p></td>
<td><p>The total number of points
that make up the spine.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[2]</p></td>
<td><p>x</p></td>
<td><p>X coordinate.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[3]</p></td>
<td><p>y</p></td>
<td><p>Y coordinate.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>[4]</p></td>
<td><p>z</p></td>
<td><p>Z coordinate.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>[5]</p></td>
<td><p>d</p></td>
<td><p>Diameter of the spine at
this coordinate.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 16 The &lt;property name=”Backbone”&gt; element, its values, and a descriptions for all fields. The Line # column corresponds to the first 5 line numbers in Figure 30. The x, y, z, and d values are repeated in lines [6] through [21] of Figure 30.</em></p>
<p>The final property, “VolumeRLE”, includes values detailing a cuboid that encloses the detected spine volume. Each value is defined in Table 17.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;VolumeRLE&quot;</span><span class="nt">&gt;&lt;s&gt;</span>x.xx y.yy z.z VT Vx Vy Vz x.x y.y z.z B1 F1 B2 F2 ... Bn Fn<span class="nt">&lt;/s&gt;&lt;/property&gt;</span>
</pre></div>
</div>
<p><em>Figure 31 The “VolumeRLE” &lt;property&gt; includes a string of numbers that describe the voxels of the spine’s detection cuboid. The variables in the figure correspond with one number, each defined in Table 17. The first two pairs of background and foreground values (B1, F1, B2, F2) are listed. The remaining background and foreground values are abbreviated in the figure for concision using the ellipsis. The final pair of background and foreground values (Bn, Fn) are the last values in the string.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 15%" />
<col style="width: 14%" />
<col style="width: 51%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>&lt;property&gt;</strong></p></th>
<th class="head"><p><strong>Figure</strong>
<strong>Variable</strong></p></th>
<th class="head"><p><strong>Value</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="3"><p>name=”VolumeRLE”</p></td>
<td><p>A list of the foreground and background
voxels that construct a cuboid
enclosing the spine volume. The
foreground voxels are the voxels within
the cuboid with a high luminance that
construct the spine. In contrast,
background voxels have a low luminance
value and are excluded from the spine
volume. The first ten values of the
child element describe the detection
cuboid. The rest of the numbers are a
series of pairs reporting the
background and foreground voxels run
lengths within the cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>x.xx</p></td>
<td><p>x scaling</p></td>
<td><p>X scaling of the image the spine was
detected from.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>y.yy</p></td>
<td><p>y scaling</p></td>
<td><p>Y scaling of the image the spine was
detected from.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>z.zz</p></td>
<td><p>z scaling</p></td>
<td><p>Z scaling of the image the spine was
detected from.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>V<sub>T</sub></p></td>
<td><p>voxel
count</p></td>
<td><p>Total number of foreground voxels.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>V<sub>x</sub></p></td>
<td><p>x voxel
count</p></td>
<td><p>Total number of voxels in the X
direction that construct the detection
cuboid.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>V<sub>y</sub></p></td>
<td><p>y voxel
count</p></td>
<td><p>Total number of voxels in the Y
direction that construct the detection
cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>V<sub>z</sub></p></td>
<td><p>z voxel
count</p></td>
<td><p>Total number of voxels in the Z
direction that construct the detection
cuboid.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>x.x</p></td>
<td><p>x origin</p></td>
<td><p>X origin of the cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>y.y</p></td>
<td><p>y origin</p></td>
<td><p>Y origin of the cuboid</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>z.z</p></td>
<td><p>z origin</p></td>
<td><p>Z origin of the cuboid.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>B<sub>1</sub></p></td>
<td><p>background
voxel
-run1</p></td>
<td><p>The number of background voxels
beginning at the cuboids defined origin
and ending at the first foreground
voxel.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>F<sub>1</sub></p></td>
<td><p>foreground
voxel
-run1</p></td>
<td><p>The number of foreground voxels between
the last reported background voxel and
the next background voxel.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 17 The &lt;property name=”VolumeRLE”&gt; element, its values, and a descriptions for all fields. The Figure Variable column corresponds to the variables in Figure 31.</em></p>
<p>The final values of the “VolumeRLE” &lt;property&gt; include a list of the foreground and background voxels that construct a cuboid enclosing the spine volume. The foreground voxels are the voxels within the cuboid with a high luminance that make up the spine’s volume. The foreground and background voxels are reported as a series of numbers indicating the voxels run lengths within the cuboid. The first run will count the number of background voxels beginning at the cuboids defined origin and ending at the first foreground voxel. The next number of the string reports the run of foreground voxels until the next background voxel is reached. The run-length moves positively in X until the cuboid boundary is reached. Next, the voxels are reported for the second line of Y until the first Z plane has been reported. The voxels are reported in the same pattern until all Z planes making up the cuboid have been reported.</p>
</div>
<div class="section" id="varicosity">
<h5>5.5.1.2 &lt;varicosity&gt;<a class="headerlink" href="#varicosity" title="Permalink to this headline">¶</a></h5>
<p>Neuronal varicosities, or boutons, are clusters of neurotransmitters that appear along neuronal trees as swellings in the branch. Since the structures are part of the neuronal tree, it is necessary to represent their relationship to the corresponding branch and point location. In the neuromorphological data file, the &lt;varicosity&gt; element is nested within the relating &lt;tree&gt; and is made up of five &lt;point&gt; elements that coalesce to form the volume of that varicosity. Each &lt;point&gt; diameters (d) describes the thickness of the varicosity at that location.</p>
<a class="reference internal image-reference" href="_images/4.png"><img alt="" class="align-center" src="_images/4.png" style="width: 750px; height: 200px;" /></a>
<p><em>Figure 32 (a) The data format for one &lt;varicosity&gt; element nested with a &lt;tree&gt;. For conciseness, the attributes of most &lt;point&gt; elements have been abbreviated with an ellipsis. (b) A diagram of a varicosity along a neuronal tree. The five points of the varicosity are represented with gray circles. Each has a thickness, d. The varicosity’s center point thickness describes the maximum diameter of the varicosity. The length of the varicosity is the distance between the first and last varicosity points. By averaging the diameters of the first and last varicosity points, the branch thickness is estimated.</em></p>
<p>Each attribute of the varicosity element is described in Table 18.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 21%" />
<col style="width: 39%" />
<col style="width: 39%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;varicosity&gt;</p></td>
<td></td>
<td><p>Varicosity elements are
nested within the
relating tree, branch,
and point location. It
contains five nested
&lt;point&gt; elements that
coalesce to form the
volume of that
varicosity.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>version</p></td>
<td><p>Defines the data format
version for the
varicosity child
element.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>color</p></td>
<td><p>Defines the color of the
varicosity, represented
with a hexadecimal RGB
value.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>generated</p></td>
<td><p>The
value “true” indicates
the varicosity was
detected by an MBF
Bioscience software
algorithm. The value
“false” indicates a
manual or alternative
detection method was
used to create the
varicosity.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>length</p></td>
<td><p>Distance between the
first varicosity point
to the last varicosity
point reported in
micrometer units</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>maximumdiameter</p></td>
<td><p>Largest diameter along
the varicosity profile</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>thicknessratio</p></td>
<td><p>The ratio of the
varicosity diameter at
the middle point and the
underlying branch
thickness.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>is2d</p></td>
<td><p>If the varicosity is
detected from a 2D
image, the value will
equal “true”. The is2d
attribute will be marked
“false” if the
varicosity is detected
from a 3D image stack.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>anchoroffset</p></td>
<td><p>Defines the position
along the tree, between
the attachment and the
previous point that
gives
the precise location of
the center of the
varicosity.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>attachment</p></td>
<td><p>Describes the point on
the branch where the
center of the varicosity
is attached.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 18 The &lt;varicosity&gt; element and its attributes. Each data component includes a corresponding description.</em></p>
</div>
</div>
<div class="section" id="somas-and-cell-bodies">
<h4>5.5.2 Somas and Cell Bodies<a class="headerlink" href="#somas-and-cell-bodies" title="Permalink to this headline">¶</a></h4>
<p>2D or 3D cell bodies are represented using the &lt;contour&gt; element or groups of &lt;contour&gt; elements. The differentiating factor between a contour and a cell body is found in the name attribute of the &lt;contour&gt;. All cell body contours have names containing “soma “(case insensitive and space relevant).  3D cell bodies are traced using multiple contours of the same name at different z positions outlining the entire Z space of the cell body region.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;contour</span> <span class="na">name=</span><span class="s">&quot;Soma N&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">closed=</span><span class="s">&quot;true&quot;</span> <span class="na">shape=</span><span class="s">&quot;Contour&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;GUID&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;FillDensity&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Channel&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  ⋮
<span class="nt">&lt;/contour&gt;</span>
<span class="nt">&lt;contour</span> <span class="na">name=</span><span class="s">&quot;Soma N&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">closed=</span><span class="s">&quot;true&quot;</span> <span class="na">shape=</span><span class="s">&quot;Contour&quot;</span><span class="nt">&gt;</span>
  ⋮
<span class="nt">&lt;/contour&gt;</span>
<span class="nt">&lt;contour</span> <span class="na">name=</span><span class="s">&quot;Soma N&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">closed=</span><span class="s">&quot;true&quot;</span> <span class="na">shape=</span><span class="s">&quot;Contour&quot;</span><span class="nt">&gt;</span>
  ⋮
<span class="nt">&lt;/contour&gt;</span>
<span class="nt">&lt;contour</span> <span class="na">name=</span><span class="s">&quot;Soma N&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">closed=</span><span class="s">&quot;true&quot;</span> <span class="na">shape=</span><span class="s">&quot;Contour&quot;</span><span class="nt">&gt;</span>
  ⋮
<span class="nt">&lt;/contour&gt;</span>
</pre></div>
</div>
<p><em>Figure 33 Multiple &lt;contour&gt; elements  named “Soma N”, indicating all &lt;contour&gt; elements make up a single cell body. The cell body &lt;contour&gt; contains the same child elements, attributes, and values of a &lt;contour&gt;. These have been abbriviated in the figure with a verticle ellipsis for the last three &lt;contour&gt; elements. The &lt;property name=” “&gt; child elements of the first &lt;contour&gt; listed exclude all values for concision. The &lt;point&gt; attributes have been abbriviated in this figure with an ellipsis. A contour usually contains a list of many &lt;point&gt; elements, connected based on the order they are listed. These have also been abbriviated in the figure with a vertical ellipsis.</em></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>It’s possible that data files may represent cell bodies with a contour name=”CellBody” and the closed attribute will equal “false”. This represents a closed cell body contour despite being marked as open.</p>
</div>
</div>
</div>
<div class="section" id="vascular-networks">
<h3>5.6 Vascular Networks<a class="headerlink" href="#vascular-networks" title="Permalink to this headline">¶</a></h3>
<div class="section" id="vessel">
<h4>5.6.1 &lt;vessel&gt;<a class="headerlink" href="#vessel" title="Permalink to this headline">¶</a></h4>
<p>Structures like vasculature or nerve bundles are be represented with a different branching structure than the trees. In the neuromorphological file format, &lt;vessel&gt; elements are made up of &lt;nodes&gt; that connect by &lt;edges&gt;. The &lt;edge&gt; is a collection of connected points. The connections relationships are described with the &lt;edgelists&gt; element.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;vessel</span> <span class="na">version=</span><span class="s">&quot;2&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">type=</span><span class="s">&quot;undirected&quot;</span> <span class="na">name=</span><span class="s">&quot;Undirected Vessel&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;property</span> <span class="na">name=</span><span class="s">&quot;Channel&quot;</span><span class="nt">&gt;&lt;/property&gt;</span>
  <span class="nt">&lt;nodes&gt;&lt;/nodes&gt;</span>
  <span class="nt">&lt;edges&gt;&lt;/edges&gt;</span>
  <span class="nt">&lt;edgelists&gt;&lt;/edgelists&gt;</span>
<span class="nt">&lt;/vessel&gt;</span>
</pre></div>
</div>
<p><em>Figure 34 The &lt;vessel&gt; element, child elements, attributes, and values as they apear in the segmentation data file. In this figure, the vessel’s child elements do not include any child elements, attributes, and values for concision.</em></p>
<p>The &lt;vessel&gt; elements can have edges that loop, whereas &lt;tree&gt; elements can only branch. The looping capability of the &lt;vessel&gt; elements is known as a graph structure and allows for the representation of biological structures such as anastomoses in vascular or neuronal networks.</p>
<p>The &lt;vessel&gt; attributes are detailed in Table 19.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;vessel&gt;</p></td>
<td><p>A graph branching
structure that allows
for looping connections
to model vasculature or
nerve bundles.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>version</p></td>
<td><p>Defines the data format
version for the
vessel child element.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>color</p></td>
<td><p>Defines the color of the
vessel, represented with
a hexadecimal RGB value.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>type</p></td>
<td></td>
</tr>
<tr class="row-even"><td></td>
<td><p>name</p></td>
<td><p>A user-defined
classification name for
a vessel.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 19 The &lt;vessel&gt; element and its attributes. Each data component includes a corresponding description.</em></p>
<p>The &lt;nodes&gt; element contains nested &lt;node&gt; elements for each node found on that vessel. Each &lt;node&gt; is given a unique ID value when it is created. The &lt;node&gt; elements also contain one nested &lt;point&gt; element defining the center of the node within the coordinate space. The format of the &lt;point&gt; elements within the &lt;node&gt; section of the data file is consistent with the &lt;point&gt; elements throughout the rest of the data file.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;nodes&gt;</span>
  <span class="nt">&lt;node</span> <span class="na">id=</span><span class="s">&quot;0&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;/node&gt;</span>
  <span class="nt">&lt;node</span> <span class="na">id=</span><span class="s">&quot;1&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;/node&gt;</span>
  ⋮
<span class="nt">&lt;/nodes&gt;</span>
</pre></div>
</div>
<p><em>Figure 35 The &lt;nodes&gt; child elment of a &lt;vessel&gt;, its child elements, attributes, and values as they apear in the segmentation data file. The &lt;nodes&gt; element includes many &lt;node&gt; child elements each with a unique ID and one  &lt;point&gt; child element. In this figure, some &lt;point&gt; and &lt;node&gt;  elements have been abriviated with ellipsis.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;nodes&gt;</p></td>
<td><p>A set of terminations
points that include the
vessel’s origin, ending,
and any branch points.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>&lt;node&gt;</p></td>
<td><p>Each termination point
is detailed in the
&lt;node&gt; element. All
&lt;node&gt; elements are
nested within the
&lt;nodes&gt; child element.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>id</p></td>
<td><p>A unique identifier is
generated for every node
within the vessel.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 20 The &lt;nodes&gt; child element of a &lt;vessel&gt;, its child elements, attributes, and values. A description is provided for all fields.</em></p>
<p>The next child element of the &lt;vessel&gt; is the &lt;edges&gt; element. It will contain a list of all the edges, or branches, that make up the traced vessel. Each branch is represented by a nested &lt;edge&gt; element which includes a unique ID value, and the collection of &lt;point&gt; elements that make up the branch.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;edges&gt;</span>
  <span class="nt">&lt;edge</span> <span class="na">id=</span><span class="s">&quot;0&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;/edge&gt;</span>
  <span class="nt">&lt;edge</span> <span class="na">id=</span><span class="s">&quot;1&quot;</span> <span class="na">type=</span><span class="s">&quot;origin&quot;</span><span class="nt">&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
    <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;d.dd&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;/edge&gt;</span>
  ⋮
<span class="nt">&lt;/edges&gt;</span>
</pre></div>
</div>
<p><em>Figure 36 The &lt;edges&gt; child elment of a &lt;vessel&gt;, its child elements, attributes, and values as they apear in the segmentation data file. The &lt;edges&gt; element includes many &lt;edge&gt; child elements each with a unique ID and a list of sequentially ordered  &lt;point&gt; child elements. In this figure, some &lt;point&gt; and &lt;edge&gt;  elements have been abriviated with ellipsis.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>&lt;edges&gt;</p></td>
<td><p>A set of edges or
branches that makes up
the vessel</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>&lt;edge&gt;</p></td>
<td><p>Each branch of the
vessel is constructed of
a collection of
sequentially ordered and
connected points.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>id</p></td>
<td><p>The unique identifier is
generated for every
edge.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 21 The &lt;edges&gt; child element of a &lt;vessel&gt;, its child elements, attributes, and values. A description is provided for all fields.</em></p>
<p>Lastly, the &lt;edgelist&gt; element holds the key for how the nodes and edges relate. Each nested &lt;edgelist&gt; has four attributes. The &lt;edgelist&gt; is given a unique ID, recorded with the id attribute. The edge attribute includes the id of the &lt;edge&gt; element (branch of the vessel) the &lt;edgelist&gt; is describing the node relationship for. The sourcenode attribute is the ID of the &lt;node&gt; that the &lt;edge&gt; starts from, where the targetnode attribute is the ID of the &lt;node&gt; that the &lt;edge&gt; ends at.  If either the sourcenode or target node values equal -1, this means that there is no starting node and this is the origin of the vessel.</p>
<a class="reference internal image-reference" href="_images/5.png"><img alt="" class="align-center" src="_images/5.png" style="width: 750px; height: 500px;" /></a>
<p><em>Figure 37 (a) The data format for the &lt;edgelists&gt; child element of a &lt;vessel&gt;. Each &lt;edgelist&gt; id attribute corresponds to the &lt;edgelist&gt; ids in Figure 37b, informing how the &lt;vessel&gt; &lt;edge&gt; elements connect to the &lt;node&gt; elements. The edge attributes correspond to the edge ids in Figure 36. The sourcenode and targetnode values refer to a node id in Figure 35. (b) A diagram of the &lt;edgelists&gt; element in Figure 37a. Each &lt;edgelist&gt; and &lt;edge&gt; id correspond to one of the vessel branches. The &lt;edgelist&gt; sourcenode and targetnode inform the start and endpoint of the vessel branch or &lt;edge&gt;. For example, edge=”4” (E4) begins at node 2 (N2) and ends at node 3 (N3). This connection is indicated in &lt;edgelist&gt; id=4 (see Figure 37a). This connection of the vessel back onto itself creates a loop structure.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 23%" />
<col style="width: 38%" />
<col style="width: 38%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;edgelists&gt;</p></td>
<td><p>A set of edgelist child
elements whose
attributes define the
relationship between an
edge and its two nodes.</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>&lt;edgelist&gt;</p></td>
<td><p>Defines the
relationship between
edge and node elements,
indicating the branches
and loops that form the
vessel.</p></td>
<td></td>
</tr>
<tr class="row-even"><td></td>
<td><p>id</p></td>
<td><p>The unique identifier
is generated for every
edgelist.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>edge</p></td>
<td><p>Indicates the ID of the
edge element that the
edgelist relates to a
start and end node.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>sourcenode</p></td>
<td><p>The ID of the node that
the edge starts from.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>targetnode</p></td>
<td><p>The ID of the node that
the edge ends at.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 22 The &lt;edgelists&gt; child element of a &lt;vessel&gt;, its child elements, attributes, and values. A description is provided for all fields.</em></p>
</div>
</div>
<div class="section" id="annotations">
<h3>5.7 Annotations<a class="headerlink" href="#annotations" title="Permalink to this headline">¶</a></h3>
<div class="section" id="arrow">
<h4>5.7.1 &lt;arrow&gt;<a class="headerlink" href="#arrow" title="Permalink to this headline">¶</a></h4>
<p>The &lt;arrow&gt; element includes two-point elements that connect to create the length of the arrow from the tip of the head to the tail end. This element can be used to draw attention to structures in the image or tracing data.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;arrow</span> <span class="na">name=</span><span class="s">&quot;Arrow&quot;</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span> <span class="na">tail=</span><span class="s">&quot;true&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;1.00&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;1.00&quot;</span><span class="nt">/&gt;</span>
<span class="nt">&lt;/arrow&gt;</span>
</pre></div>
</div>
<p><em>Figure 38  The &lt;arrow&gt; element, child elements, attributes, and values as they apear in the segmentation data file. The &lt;point&gt; coordinates represent the location of the head and tail of the arrow. Note that the diameter (d) of an &lt;arrow&gt; &lt;point&gt; is always set to “1.00”.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;arrow&gt;</p></td>
<td><p>A trace element with two
points connecting to
form a line with an
arrowhead. Used to draw
attention to structures
in the image or tracing.</p></td>
<td></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>name</p></td>
<td><p>The name value of an
arrow will always be
“Arrow”.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>color</p></td>
<td><p>Defines the color of the
arrow, represented with
a hexadecimal RGB value.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>tail</p></td>
<td><p>If the value is true,
the arrow includes a
head and a tail. If the
value is false, the
arrow only has a head.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 23 The &lt;arrow&gt; element and its attributes along with a description for all fields.</em></p>
</div>
<div class="section" id="text">
<h4>5.7.2 &lt;text&gt;<a class="headerlink" href="#text" title="Permalink to this headline">¶</a></h4>
<p>User-defined text can be added to any location of a data file with the &lt;text&gt; element. This element has one &lt;point&gt; child element indicating the location of the center of the text string.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;text</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;font</span> <span class="na">name=</span><span class="s">&quot;Times New Roman&quot;</span> <span class="na">size=</span><span class="s">&quot;n&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;0.00&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;value&gt;</span>ExampleText<span class="nt">&lt;/value&gt;</span>
<span class="nt">&lt;/text&gt;</span>
</pre></div>
</div>
<p><em>Figure 39 The &lt;text&gt; element, child elements, attributes, and values as they apear in the segmentation data file. The &lt;point&gt; coordinates represent the location of the center of the text box. Note that the diameter (d) of the &lt;text&gt; &lt;point&gt; is always set to “0.00”.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;text&gt;</p></td>
<td><p>A user-defined text
added to any location of
the data file.</p></td>
<td></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>color</p></td>
<td><p>Defines the color of the
text, represented with a
hexadecimal RGB value.</p></td>
</tr>
<tr class="row-even"><td><p>&lt;font&gt;</p></td>
<td></td>
<td><p>Stores the typeface and
point size of the text
string.</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>name</p></td>
<td><p>Typeface or font name.</p></td>
</tr>
<tr class="row-even"><td></td>
<td><p>size</p></td>
<td><p>Text point size.</p></td>
</tr>
<tr class="row-odd"><td><p>&lt;value&gt;</p></td>
<td></td>
<td><p>The user-defined text
string.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 24 The &lt;text&gt; element, its child elements, attributes, and values. A description is provided for all fields.</em></p>
</div>
<div class="section" id="scalebar">
<h4>5.7.3 &lt;scalebar&gt;<a class="headerlink" href="#scalebar" title="Permalink to this headline">¶</a></h4>
<p>The &lt;scalebar&gt; element stores information on a 2D scalebar set to a user-defined length including color, location, and length in micrometer units.</p>
<div class="highlight-xml notranslate"><div class="highlight"><pre><span></span><span class="nt">&lt;scalebar</span> <span class="na">color=</span><span class="s">&quot;#RRGGBB&quot;</span><span class="nt">&gt;</span>
  <span class="nt">&lt;point</span> <span class="na">x=</span><span class="s">&quot;x.xx&quot;</span> <span class="na">y=</span><span class="s">&quot;y.yy&quot;</span> <span class="na">z=</span><span class="s">&quot;z.zz&quot;</span> <span class="na">d=</span><span class="s">&quot;0.00&quot;</span><span class="nt">/&gt;</span>
  <span class="nt">&lt;value&gt;</span>nnn<span class="nt">&lt;/value&gt;</span>
  <span class="nt">&lt;showlabel&gt;</span>true<span class="nt">&lt;/showlabel&gt;</span>
  <span class="nt">&lt;showunits&gt;</span>true<span class="nt">&lt;/showunits&gt;</span>
<span class="nt">&lt;/scalebar&gt;</span>
</pre></div>
</div>
<p><em>Figure 40 The &lt;scalebar&gt; element, child elements, attributes, and values as they apear in the segmentation data file. The &lt;point&gt; coordinates represent the location of the center of the scale bar. Note that the diameter (d) of the &lt;text&gt; &lt;point&gt; is always set to “0.00”. This &lt;scalebar&gt; element has &lt;showlabel&gt; and &lt;showunits&gt; set to true indicating the length value of the scale bar and the units are displayed alongside the scale bar.</em></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 24%" />
<col style="width: 38%" />
<col style="width: 38%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p><strong>Element</strong></p></th>
<th class="head"><p><strong>Attribute</strong></p></th>
<th class="head"><p><strong>Description</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;scalebar&gt;</p></td>
<td><p>A 2D scalebar set to a
user-defined length
based on the X and Y
pixel scaling.</p></td>
<td></td>
</tr>
<tr class="row-odd"><td></td>
<td><p>color</p></td>
<td><p>Defines the color of
the scalebar,
represented with a
hexadecimal RGB value.</p></td>
</tr>
<tr class="row-even"><td><p>&lt;value&gt;</p></td>
<td></td>
<td><p>Length of the scalebar
(µm). Must be a
positive integer.</p></td>
</tr>
<tr class="row-odd"><td><p>&lt;showlabel&gt;</p></td>
<td></td>
<td><p>If true, the numerical
&lt;value&gt; of the scalebar
is displayed. If false,
the &lt;value&gt; is hidden.</p></td>
</tr>
<tr class="row-even"><td><p>&lt;showunits&gt;</p></td>
<td></td>
<td><p>If true, the units (µm)
of the scalebar are
displayed. If false,
the units are hidden.</p></td>
</tr>
</tbody>
</table>
<p><em>Table 25 The &lt;scalebar&gt; element, its child elements, attributes, and values. A description is provided for all fields.</em></p>
</div>
</div>
</div>
<div class="section" id="references">
<h2>6. References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li><p>Extensible Markup Language (XML) 1.0 (Fifth Edition) <a class="reference external" href="https://www.w3.org/TR/2008/REC-xml-20081126/">https://www.w3.org/TR/2008/REC-xml-20081126/</a></p></li>
<li><p>ISO/IEC 885901:1998 Information technology- 8-bit single byte coded graphic character sets <a class="reference external" href="https://www.iso.org/obp/ui/#iso:std:iso-iec:8859:-1:ed-1:v1:en">https://www.iso.org/obp/ui/#iso:std:iso-iec:8859:-1:ed-1:v1:en</a></p></li>
<li><p>Allen Mouse Brain Atlas Lein, E.S. et al. (2007) Genome-wide atlas of gene expression in the adult mouse brain, Nature 445: 168-176. doi:10.1038/nature05453</p></li>
</ol>
<dl class="footnote brackets">
<dt class="label" id="id3"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p>The version number given here is for the neuromorphological file specification and is independent of the version number for the Neuromophological file format and any MBF Bioscience Software. The date after the version number is the last modification date of this document.</p>
</dd>
</dl>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Release.html" class="btn btn-neutral float-right" title="Release Notes" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="Welcome to the Neuromorphological File Specification" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, MBF Bioscience

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>